<!doctype html>
<html class="no-js" lang="en">
<head>
<meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />

<title>Erik Amézquita &mdash; A hodge-podge of math and plants</title>
<meta name="description" content="Erik Amézquita webpage">

<!-- ******************************************************************************************
Set the type and color theme here - the Pro version contains additional themes -->
<link href="../css/hawthorne_type1_color2.css" rel="stylesheet">
<!--  ************************************************************************************* -->

<link href="../css/font-awesome.min.css" rel="stylesheet">
<script src="../js/vendor/modernizr.js"></script>

</head>
<body>

<div class="row">
    <div class="small-6 medium-12 large-12 small-centered columns">
        <header>
            <h1><a href="../index.html">Erik Amézquita</a></h1>
            <h2><a href="../index.html">PFFIE Postdoctoral Fellow</a></h2>
            <h2><a href="../index.html">Plant Sciences and Mathematics @ University of Missouri</a></h2>
            <!--
            <div class="logo">
                <a href="img/front_original.jpg"><img src="img/front_medium.jpg" alt="Erik_in_Chicago_2019" /></a>
            </div>
            -->
        </header>
    </div>
    <div class="small-6 medium-12 large-12 small-centered columns">
        <nav>
            <ul class="inline-list-custom">
                <li><a href="../index.html">Home [TL;DR]</a></li>
                <li><a href="../research.html">Research</a></li>
                <li><a href="../teaching.html">Academic Resources</a></li>
                <li><a href="../service.html">Other Resources</a></li>
                <li><a href="../aboutme.html">About Me</a></li>
                <li><a href="../else.html">Else</a></li>
            </ul>
        </nav>
    </div>
</div>

<div class="row">
	<div class="small-12 medium-7 large-7 columns">
		<h2>The shape of things to come</h2>
		<p style="text-align:right">
            <em>(Taken from the introduction of my doctoral dissertation)</em>
        </p>
        <h3>Morphometrics background and motivation</h3>
        <p>
            Shape plays a fundamental role across all organisms at all observable levels. Molecules and proteins constantly fold and wrap into intricate designs inside our cells. Cells arrange into elaborate motifs to form sophisticated tissues. Layers of different tissues come together to form delicate vascular and nervous systems that sustain hands, wings, or fins. Each of these limbs evolved as part of a distinct branch of the ever-growing tree of life. From micro-biology to macro-evolutionary scales, shape and its patterns are foundational to biology. Measuring and understanding the shape is key to extracting valuable information from data, and push further our insights. 
        </p>
        <p>
            Even if we limit our scope to plant biology, a simple glance outdoors reveals a large diversity of shape among flowers, leaves, fruits, and branches. A first attempt to characterize the shape of a plant could use traditional morphometrics, describing the shape of the plant in terms of height, stem thickness, or number of branches. 
            With these measures at hand, we could look for allometry &mdash;the relative growth of parts of an organism to the whole&mdash; and thus linearly transform biological shapes between each other.
            However, any given plant shape is too complex to measure it simply in terms of length, width, and branching angles.
        </p>
        <p>
            We could use 
            <a href="https://en.wikipedia.org/wiki/Morphometrics#Landmark-based_geometric_morphometrics">Geometric Modern Morphometrics (GMM)</a> instead, where we first define homologous landmark points on every sample and then measure shape similarity by overlapping all these landmarks and computing their Euclidean differences.
            The computation of differences can be refined by rotating, translating, and scaling appropriately the landmark coordinates prior to the general overlap. This procedure, known as generalized Procrustes analysis, defines a morpho-space, or a space of all possible shapes based on all the possible landmark configurations, which allows us to define overall shape distance. The GMM approach can produce distorted results if there are not enough landmarks shared across all sampled individuals, which could occur if we attempt to compare tissues from different families.
        </p>
        <div class="row" style="text-align:center; font-size: 0.1em; color: Grey;">
                <img src="../barley/figs/chitwood_2021_fig-004_1.jpg" alt="normal"  style="float: center; height: 300px; margin: 0 0"/>
                <img src="../barley/figs/chitwood_sinha_2016_01_02.jpg" alt="original"  style="float: center; height: 300px; margin: 0 0"/>
        </div>
        <p style="text-align:center; font-size: 0.8em; color: Grey;">
            Credits: <a href=" https://doi.org/10.1002/ppp3.10157">Chitwood  (2020)</a>
            and
            <a href="https://doi.org/10.1016/j.cub.2016.02.033">Chitwood and Sinha (2016)</a>
        </p>
        <p>
            In the absence of corresponding sets of coordinates, we may attempt to describe the outline of the 
            <a href="https://en.wikipedia.org/wiki/Morphometrics#Outline_analysis">shape using Fourier analysis</a>, by considering the outline as a harmonic series, or the sum of wave-like curves. All of the above strategies, based on either morphometrics or Fourier analysis, have proven to be extremely insightful to uncover hidden patterns that mold diverse organism shapes at genetic, developmental, evolutionary, and environmental levels. However, it is unclear how to extend this technique beyond 2D images.
        </p>
        <p>
            Inspired by the inherent difficulties of working with 3D data, specifically <a href="https://www.youtube.com/@endlessforms6756">X-ray CT scan imaging data</a>,
            we turn to Topological Data Analysis (TDA).
        </p>
        <p style="text-align:center">
            <iframe width="512" height="300" src="https://www.youtube-nocookie.com/embed/oM9kAq0PBvw" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
        </p>
        <h3>Topological Data Analysis (TDA)</h3>
        <p>
            Topological Data Analysis (TDA) is a combination of different mathematical and computational disciplines that seeks to describe concisely and comprehensively the shape of data in a general setting. 
            In extremely succinct terms, TDA consists of two basic ingredients and a key idea. 
        </p>
        <div class="list">
            <ul>
                <li><strong>1st Ingredient</strong>: Think of the data as a collection of points, lines, faces, cubes, etc.</li>
                <li><strong>2nd Ingredient</strong>: Define a notion of distance between every pair of points.</li>
                <li><strong>Key Idea</strong>: Keep track of distinct shape features that form and disappear as we connect and merge several points: connected components, holes, and voids.</li>
            </ul>
        </div>
        <p style="text-align:center;font-size: 0.8em; color: Grey;">
            <img src="https://anatomypubs.onlinelibrary.wiley.com/cms/asset/9e0d7fe5-2ede-41fa-8232-ccdb8d91017f/dvdy175-fig-0006-m.jpg" alt="Ellipsoid Projection" width="80%"/>
        </p>
        <p>
            The points could be atoms, biomolecultes, cells' nuclei, image pixels, or an organism itself. Distances between points could be the Euclidean, geodesic, genetic, or correlation-based. Once we have data points and distances, known formally as a metric space, we can connect these points starting with those that are closer to each other first.
            These ingredients and idea, albeit simple, are extremely versatile and can be adapted to a myriad of contexts and data collections. Moreover, the notion of shape presented by TDA is limited solely by the data itself, unleashing it from possible selection biases. This very adaptability and impartiality makes TDA a powerful data analysis tool that can further our insights in a variety of plant biology scenarios.
        </p>
        <h3>Plenty of TDA applications in the biology literature</h3>
        <div class="row" style="text-align:center">
            <div class="column" style="max-width:45%">
                <p><strong>Histological image processing</strong></p>
                <img style="padding: 0 0 0 0;" src="https://ars.els-cdn.com/content/image/1-s2.0-S1361841518302688-gr1.jpg">
                <img style="padding: 0 0 0 0;" src="https://ars.els-cdn.com/content/image/1-s2.0-S1361841518302688-gr2.jpg">
                <p style="font-size: 0.8em; text-align: right; color: Grey;"> Credits: <a href="https://doi.org/10.1016/j.media.2019.03.014">Qaiser <em>et al.</em> (2019)</a></p>
            </div>
            <div class="column" style="max-width:45%">
                <p><strong>HIV phylogenetics</strong></p>
                <img style="padding: 0 0 0 0;" src="https://www.pnas.org/cms/10.1073/pnas.1313480110/asset/ce30e9df-d595-4520-a68a-74f3c6f151d1/assets/graphic/pnas.1313480110fig01.jpeg">
                <img style="padding: 0 0 0 0;" src="https://www.pnas.org/cms/10.1073/pnas.1313480110/asset/b733c3a3-00f2-43d5-b372-9da25f3d33c9/assets/graphic/pnas.1313480110fig02.jpeg">
                <p style="font-size: 0.8em; text-align: right; color: Grey;"> Credits: <a href="https://doi.org/10.1073/pnas.1313480110">Chan <em>et al.</em> (2013)</a></p>
            </div>
        </div>
        <div class="row">
            <div class="column" style="width:50%;">
                <p style="text-align: center;">Holes</p>
                <p style="text-align: center;">&darr;</p>
                <p style="text-align: center;">Cancerous tissue</p>
            </div>
            <div class="column" style="width:50%;">
                <p style="text-align: center;">Holes</p>
                <p style="text-align: center;">&darr;</p>
                <p style="text-align: center;">Horizontal Reassortment</p>
            </div>
        </div>
        <div class="row" style="text-align:center">
            <div class="column" style="max-width:45%">
                <p><strong>Protein structures</strong></p>
                <img style="padding: 0 0 0 0;" src="https://www.degruyter.com/document/doi/10.1515/sagmb-2015-0057/asset/graphic/j_sagmb-2015-0057_fig_001.jpg">
                <img style="padding: 0 0 0 0;" src="https://www.degruyter.com/document/doi/10.1515/sagmb-2015-0057/asset/graphic/j_sagmb-2015-0057_fig_012.jpg">
                <img style="padding: 0 0 0 0;" src="https://www.degruyter.com/document/doi/10.1515/sagmb-2015-0057/asset/graphic/j_sagmb-2015-0057_fig_010.jpg">
                <p style="font-size: 0.8em; text-align: right; color: Grey;"> Credits: <a href="https://doi.org/10.1515/sagmb-2015-0057">Kovacev-Nikolic <em>et al.</em> (2016)</a></p>
            </div>
            <div class="column" style="max-width:50%">
                <p><strong>3D imaging of panicles</strong></p>
                <img style="padding: 0 0 0 0;" src="https://media.springernature.com/original/springer-static/image/chp%3A10.1007%2F978-3-030-20867-7_7/MediaObjects/484957_1_En_7_Fig1_HTML.png">
                <p style="font-size: 0.8em; text-align: right; color: Grey;"> Credits: <a href="https://doi.org/10.1007/978-3-030-20867-7_7">Chitwood <em>et al.</em> (2019)</a></p>
            </div>
        </div>
        <div class="row">
            <div class="column" style="width:45%;">
                <p style="text-align: center;">Holes</p>
                <p style="text-align: center;">&darr;</p>
                <p style="text-align: center;">Open/closed conformations</p>
            </div>
            <div class="column" style="width:50%;">
                <p style="text-align: center;">Components</p>
                <p style="text-align: center;">&darr;</p>
                <p style="text-align: center;">Panicle structure</p>
            </div>
        </div>
        <p class="back-to-top-holder"><a class="back-to-top"><i class="fa fa-chevron-circle-up fa-3x"></i></a></p>
	</div>
	<div class="small-12 medium-5 large-5 columns">
		<p>
            ¡<strong>Published review</strong>: Amézquita <em>et al.</em> (2020)!
        </p>
        <div class="thumbnail-img">
            <div class="thumbnail-hover"><a href="https://doi.org/10.1002/dvdy.175"></a></div>
            <a href="https://doi.org/10.1002/dvdy.175" target="_blank"><img src="figs/amezquita_etal_2020.png" alt="Amezquita et al 2020" style="vertical-align:bottom"/></a>
        </div>
        <p>
            DOI: <a href="https://doi.org/10.1002/dvdy.175" target="_blank">10.1002/dvdy.175</a>
        </p>
        <p>
            &mdash;
        </p>
        <p>
            <strong><a href="slides/mizzou_2023.html" target="_blank">As slides</a></strong>: 
            Presented at the <a href="https://math.missouri.edu/seminars" target="_blank">Geometry and Topology Seminar</a>. Department of Mathematics. University of Missouri, Columbia, Missouri. October 2023.
        </p>
        <p>
            &mdash;
        </p>
        <p>
            <img src="figs/vr1_07.jpg" alt="Amezquita et al 2021" style="vertical-align:bottom"/>
        </p>
        <p>
            <img src="figs/mapper_b_complete.jpg" alt="Amezquita et al 2021" style="vertical-align:bottom"/>
        </p>
        <p>
            &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
        </p>
        <p style="font-size:1.5em">
            <strong><a href="#additional_resources">Go to additional resources</a></strong>
        </p>
        <p class="back-to-top-holder"><a class="back-to-top"><i class="fa fa-chevron-circle-up fa-3x"></i></a></p>
        <h3>Other research projects</h3>
        <div class="list">
            <ul>
                <li><a href="../tda/description.html" class="current">The intersection of Topological Data Analysis and Biology</a></li>
                <li><a href="../cuscuta/description.html">The early dodder gets the host</a></li>
                <li><a href="../walnuts/description.html">The crackability of walnuts: all about shape, in a nutshell</a></li>
                <li><a href="../barley/description.html">Quantification of barley grain morphology</a></li>
                <li><a href="../journal_diversity/description.html">Global disparities in plant biology research</a></li>
                <li><a href="../nasrin/description.html">Mapper to unravel the shape of omics data</a></li>
                <li><a href="../citrus/description.html">The shape of citrus fruits and modeling their oil gland distribution</a></li>
                <li><a href="../demat/description.html">Archaeological artifact classification and the Euler Characteristic</a></li>
            </ul>
        </div>
	</div>
</div>

<div class="row">
    <div class="small-12 medium-12 large-12 columns">
        <footer>
            <ul>
                <li>&copy; 2023 <a href="index.html">Erik Amézquita</a>. All rights reserved.</li>
                <li>Last updated: September 2023.</li>
                <li><a href="mailto:amezqui3@msu.edu"><i class="fa fa-envelope-o"></i> eah4d_at_missouri_dot_edu</a></li>

                <!-- *****************************************************************************************
                The "Powered by Type & Grids" link in the footer is required in the Free version
                typeandgrids.com/pirates -->
                <li>Powered by Type &amp; Grids [The link to the template was broken last time I checked]</li>
                <!--  ************************************************************************************ -->

            </ul>
            <div class="social-icons">
                <a href="https://github.com/ejamezquita/" target="_blank"><i class="fa fa-github-square fa-2x"></i></a>
                <a href="https://www.linkedin.com/in/erik-amezquita/" target="_blank"><i class="fa fa-linkedin-square fa-2x"></i></a>
                <a href="https://www.researchgate.net/profile/Erik-Amezquita-2" target="_blank"><img src="../img/researchgate.svg" width="30" alt="researchgate" style="vertical-align:bottom; margin: 0px 10px"/></a>
                <a href="https://orcid.org/0000-0002-9837-0397" target="_blank"><img src="../img/orcid.svg" width="35" alt="researchgate" style="vertical-align:bottom"/></a>
                <a href="https://scholar.google.com/citations?user=JBCDR4cAAAAJ&hl=en" target="_blank"><img src="../img/googlescholar.svg" width="35" alt="researchgate" style="vertical-align:bottom"/></a>
            </div>
        </footer>
    </div>
</div>

<script src="../js/vendor/jquery.min.js"></script>
<script src="../js/foundation.min.js"></script>
<script src="../js/hawthorne.js"></script>

</body>
</html>
