<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Characterizing transcriptome spatial patterns in soybean tissues</title>
    <meta charset="utf-8" />
    <meta name="author" content="Sutton Tennant, Erik Amézquita, Sandra Thibivillers, Marc Libault - Division of Plant Science &amp; Technology Department of Mathematics University of Missouri -" />
    <meta name="date" content="2024-05-30" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../css/mu.css" type="text/css" />
    <link rel="stylesheet" href="../../css/gallery.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Characterizing transcriptome spatial patterns in soybean tissues
]
.subtitle[
## Topological Data Analysis and molecular cartography
]
.author[
### Sutton Tennant, <strong>Erik Amézquita</strong>, Sandra Thibivillers, Marc Libault<br>-<br>Division of Plant Science &amp; Technology<br>Department of Mathematics<br>University of Missouri<br>-
]
.date[
### 2024-05-30
]

---




background-image: url("../figs/D2_molecular_cartography.jpg")
background-size: 600px
background-position: 99% 99%

# Molecular cartography

.left-column[
- 3.8M individual transcripts
- Coordinates `\((x, y, z)\)` for each of them
- Transcripts corresponding to 97 genes
- ~2400 cells in the nodule cross section
- Cross-sections from nodule and root soybean tissues
]

---

# Variety of transcript distribution

![](../figs/D2_distribution_example.jpg)

- Different cells, different shapes and sizes
- Beyond density: How to quantify and compare patterns?
- Patterns across the whole cross section? Patterns within cells?

---

class: inverse, middle, center

# Kernel Density Estimators (KDEs)

## The continuous version of a histogram

### Think of heatmaps

---

# Say we want to characterize the distribution of these points in 1D

![](../../tda/figs/kernel_density_estimator_00.svg)

- We only know the samples (blue points)

---

# A histogram gets us a sense of distribution

![](../../tda/figs/kernel_density_estimator_01.svg)

- We only know the samples (blue points)

---

# We usually read histograms as frequency tables

![](../../tda/figs/kernel_density_estimator_02.svg)

---

# But these histograms can also represent distribution densities

![](../../tda/figs/kernel_density_estimator_03.svg)

- The total gray area equals 1
- 100% of the points are represented in the histogram

---

# It is hard to do much with the histograms

![](../../tda/figs/kernel_density_estimator_04.svg)

- A continuous approximation is mathematically better to perform meaningful statistics
- Kernel Density Estimate: KDE

---

# The width/number of bins does influence the shape of the histogram

![](../../tda/figs/kernel_density_estimator_05.svg)

- Similarly, we can control the bandwidth parameter of the KDE to influence its shape
- Plenty of heuristics to define the "right" bandwidth
- But ultimately, "right" depends on the application in mind

---

# How to combine multiple KDEs?

![](../../tda/figs/kernel_density_estimator_06.svg)

---

# How to combine multiple KDEs?

![](../../tda/figs/kernel_density_estimator_07.svg)

- The area below each curve must equal 1

---

# How to combine multiple KDEs?

![](../../tda/figs/kernel_density_estimator_08.svg)

- Normalization depends on the total number of samples considered each time


---


class: inverse, middle, center

# Kernel Density Estimators (KDEs)

## The continuous version of a histogram

### Think of heatmaps

---

## KDEs: One per cell; reflect borders and nuclei


&lt;img src="../figs/D2_GLYMA_05G092200_1767_1765_kde_correction.jpg" width="700px" style="display: block; margin: auto;" /&gt;&lt;img src="../figs/D2_GLYMA_05G092200_1749_1748_kde_correction.jpg" width="700px" style="display: block; margin: auto;" /&gt;

---

# Normalize: What is 100%?

- Limit scope to just the 968 infected cells

- **By gene:** The sum of all transcripts of a fixed gene add to 100%
    - Each gene adds to 100% independent of the rest
    - Compares relative concentrations 

- **By cell AND gene:** The sum of all transcripts OF ALL genes add to 100%
    - Each gene adds to a certain percentage
    - The sum of all genes add to 100%
    - Compares absolute concentrations 

```
gene
GLYMA_17G195900    698882
GLYMA_05G216000     12919
GLYMA_17G185800     11462
GLYMA_05G203100      4116
GLYMA_15G210100     11688
GLYMA_05G092200    479091
```

---

class: inverse, middle, center

# Quantify the shape of these heatmaps

## Detour into Topological Data Analysis (TDA) and statistical learning

---



# Superlevel persistence

![](../../tda/figs/suplevel_filtration_00.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_01.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_02.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_03.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_04.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_05.svg)

---
# Superlevel persistence

![](../../tda/figs/suplevel_filtration_06.svg)


---

# Same idea for higher dimensions

![](https://pubs.acs.org/cms/10.1021/acs.jpcb.1c00904/asset/images/large/jp1c00904_0001.jpeg)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt;Credits: &lt;a href="https://doi.org/10.1021/acs.jpcb.1c00904"&gt;Manuchehrfar &lt;em&gt;et al.&lt;/em&gt; (2022)&lt;/a&gt;&lt;/p&gt;

- KDEs and superlevel persistence can be computed easily for 2D and 3D data (and even higher dimensions)

---



## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_17G195900_pd_suplevel_by_both_00557.jpg)

---
## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_05G216000_pd_suplevel_by_both_00557.jpg)

---
## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_17G185800_pd_suplevel_by_both_00557.jpg)

---
## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_05G203100_pd_suplevel_by_both_00557.jpg)

---
## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_15G210100_pd_suplevel_by_both_00557.jpg)

---
## Get a topological signature for every cell and gene

![](../figs/D2_GLYMA_05G092200_pd_suplevel_by_both_00557.jpg)


---

background-image: url("../figs/D2_GLYMA_05G216000_pd_suplevel_by_gene_01726.jpg")
background-size: 530px
background-position: 95% 42%

## Normalization matters when the cell has few transcripts

&lt;img src="../figs/D2_GLYMA_05G216000_pd_suplevel_by_both_01726.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Normalizing by gene or by cell-gene produces KDEs of different heights

- These height differences are reflected in differences in persistence diagrams

- The diagram is essentially stretched out

---

background-image: url("../figs/D2_GLYMA_15G210100_pd_suplevel_by_gene_00557.jpg")
background-size: 530px
background-position: 95% 42%

## Normalization matters when the cell has few transcripts

&lt;img src="../figs/D2_GLYMA_15G210100_pd_suplevel_by_both_00557.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Normalizing by gene or by cell-gene produces KDEs of different heights

- These height differences are reflected in differences in persistence diagrams

- The diagram is essentially stretched out

---

background-image: url("../figs/D2_GLYMA_05G216000_pd_suplevel_by_gene_00557.jpg")
background-size: 530px
background-position: 95% 42%

## Normalization matters when the cell has few transcripts

&lt;img src="../figs/D2_GLYMA_05G216000_pd_suplevel_by_both_00557.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Normalizing by gene or by cell-gene produces KDEs of different heights

- These height differences are reflected in differences in persistence diagrams

- The diagram is essentially stretched out

---

background-image: url("../figs/D2_GLYMA_05G092200_pd_suplevel_by_gene_00579.jpg")
background-size: 530px
background-position: 95% 42%

## Not so much for genes with large number of transcripts

&lt;img src="../figs/D2_GLYMA_05G092200_pd_suplevel_by_both_00579.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Look at the axes values: the diagram is essentially the same

---

background-image: url("../figs/D2_GLYMA_05G092200_pd_suplevel_by_gene_01692.jpg")
background-size: 530px
background-position: 95% 42%

## Not so much for genes with large number of transcripts

&lt;img src="../figs/D2_GLYMA_05G092200_pd_suplevel_by_both_01692.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Look at the axes values: the diagram is essentially the same

---

background-image: url("../figs/D2_GLYMA_05G092200_pd_suplevel_by_gene_01726.jpg")
background-size: 530px
background-position: 95% 42%

## Not so much for genes with large number of transcripts

&lt;img src="../figs/D2_GLYMA_05G092200_pd_suplevel_by_both_01726.jpg" width="530px" style="display: block; margin: auto auto auto 0;" /&gt;

- Normalizing by gene or by cell-gene produces KDEs of different heights

- These height differences are reflected in differences in persistence diagrams

---

# Wasserstein 1-distance

&lt;img src="../../tda/figs/bottleneck_diagram.svg" width="525px" style="display: block; margin: auto;" /&gt;

- Measure the topological similarity between two persistence diagrams `\(D_1,D_2\)`.

- Of all possible ways to match the points from `\(D_1\)` to `\(D_2\)` (or match to the diagonal), we want the one that minimizes the total sum of matching distances.

- In the caricature above, **B** is a much better matching than **A**.

- Mathematically speaking, the Wasserstein 1-distance is defined as:

`$${W_1}(D_1,D_2) = \inf_{\gamma:D_1\to D_2}\left(\sum_{p\in D_1}\left\|p - \gamma(p)\right\|_{\infty}\right)$$`

---

# Wasserstein 1-distance

.pull-left[
![](../figs/D2_ws1d_infty_suplevel_by_both_01726.svg)
]

.pull-right[
![](../figs/D2_ws1d_infty_suplevel_by_both_00579.svg)
]

- Measure the topological difference of transcript distribution of different genes within the same cell

- Comparing two genes at a time

---

# Normalization can affect the differences

.pull-left[
![](../figs/D2_ws1d_infty_suplevel_by_both_01692.svg)
]

.pull-right[
![](../figs/D2_ws1d_infty_suplevel_by_gene_01692.svg)
]

---

class: center, inverse, middle

# Normalizing by **both** cell and gene

## The sum of all transcripts of all genes of all infected cells add to 100%

The heatmaps corresponding to genes with low amounts of transcripts e.g. (`GLYMA_15G210100`) are essentially flattened and there's little than can be said about them

---

# Measure pair-wise differences

- For 6 genes, there are 15 possible different pairs

- Measure the Wasserstein distance within the same cell

- Unsurprisingly, 4 of them are quite close to each other because they are all flat.

- The most interesting differences happen between `GLYMA_05G092200` and `GLYMA_17G165900`

![](../figs/infected24_ws1d_infty_distribution_suplevel_by_both.svg)

---

# Represent spatially these distances

- Taking the median difference of each pair of genes, we can represent these distance differences in 2D with an MDS (multidimensional scaling)

.pull-left[
![](../figs/infected24_ws1d_infty_mds_suplevel_by_both.svg)
]

.pull-right[
![](../figs/D2_ws1d_infty_difference_suplevel_by_both.jpg)
]

- Darker colors represent lower difference values

---




background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01419.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01419.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01419.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 31.33\)` [1/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_02431.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_02431.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_02431.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 35.36\)` [2/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_00885.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_00885.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00885.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 38.44\)` [3/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_02063.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_02063.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_02063.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 39.77\)` [4/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_00609.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_00609.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00609.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 41.01\)` [5/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01306.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01306.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01306.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 343.77\)` [484/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_00512.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_00512.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00512.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 343.96\)` [485/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01829.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01829.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01829.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 343.99\)` [486/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_00942.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_00942.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00942.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 344.1\)` [487/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01328.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01328.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01328.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 345.72\)` [488/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01702.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01702.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01702.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 1080.96\)` [964/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_02404.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_02404.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_02404.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 1142.89\)` [965/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_02288.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_02288.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_02288.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 1171.87\)` [966/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01164.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01164.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01164.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 1251.22\)` [967/968]


---
background-image: url("../figs/D2_ws1d_infty_suplevel_by_both_01726.svg")
background-size: 200px
background-position: 99% 55%

## But how do these distributions really look like?

&lt;img src="../figs/D2_GLYMA_17G195900_z_kde_pd_suplevel_by_both_01726.jpg" width="650px"&gt;&lt;/img&gt;
&lt;br&gt;
&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_01726.jpg" width="650px"&gt;&lt;/img&gt;

`\(W_1(\text{GLYMA_17G195900, GLYMA_05G092200}) = 1532.61\)` [968/968]



---

# Some correlations

![](../figs/D2_ws1d_infty_spearman_suplevel_by_both.jpg)

---

class: inverse

# Thank you!

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:23%; font-size: 15px;"&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://bondlsc.missouri.edu/wp-content/uploads/2023/09/53236203232_6f6a6cf3cb_c-e1696541660151-370x371.jpg"&gt;
    &lt;p style="text-align: center;"&gt;Marc Libault&lt;/p&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://bondlsc.missouri.edu/wp-content/uploads/2023/09/53236204247_c28b052604_c-e1697731257686-370x316.jpg"&gt;
    &lt;p style="text-align: center";&gt;Sutton Tennant&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="width:6%; font-size: 24px;"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%; font-size: 24px; line-height:1.25"&gt;
  &lt;p style="text-align: center;"&gt;&lt;strong&gt;Email&lt;/strong&gt;&lt;/p&gt;
  &lt;p style="text-align: center;color:Blue"&gt;eah4d@missouri.edu&lt;/p&gt;
  &lt;p style="text-align: center;"&gt;&lt;strong&gt;Website and slides&lt;/strong&gt;&lt;/p&gt;
  &lt;p style="text-align: center;color:Blue"&gt;ejamezquita.github.io&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="../../js/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
