<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Shape is data, and data is shape</title>
    <meta charset="utf-8" />
    <meta name="author" content="Erik Amézquita — Division of Plant Science &amp; Technology Department of Mathematics University of Missouri —" />
    <meta name="date" content="2025-04-07" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../css/mu.css" type="text/css" />
    <link rel="stylesheet" href="../../css/gallery.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Shape is data, and data is shape
]
.subtitle[
## Mathematically phenotyping shapes and patterns<br>From molecules to organisms
]
.author[
### Erik Amézquita<br>—<br>Division of Plant Science &amp; Technology<br>Department of Mathematics<br>University of Missouri<br>—
]
.date[
### 2025-04-07
]

---




background-image: url("../../demat/figs/fam9_3.png")
background-size: 100px
background-position: 98% 2%

# About me: From MX to MI to MO at MU

## I work across multiple disciplines and countries

.left-column[
![](../../img/dvdy216-toc-0001-m.jpg)
]

.right-column[
- 2013 - 2018 : Licenciatura (Bachelor): Mathematics @ Universidad de Guanajuato and CIMAT. Thesis focused on Topological Data Analysis applied to archaeology.

- 2018 - 2023 : PhD: Computational Mathematics, Science, and Engineering @ Michigan State University. Dissertation: Exploring the mathematical shape of plants. **Came for the math. Stayed for the plants.**

- 2023 - Present : PFFIE Postdoctoral Fellow @ Division of Plant Science &amp; Technology (80%) / Department of Mathematics (20%) at MU.
]

&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;

---

# My work: Crossing and merging bridges

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:21%"&gt;
    &lt;a href="https://doi.org/10.1002/ppj2.20095" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../walnuts/figs/meat_2_05.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%"&gt;
    &lt;a href="https://doi.org/10.1007/s00299-024-03337-1" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../cuscuta/figs/4pm_rep7_plant2_v09_0401_h.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:21%"&gt;
    &lt;a href="https://doi.org/10.1002/ppj2.20095" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/Infected_Cells_05G216000_red_05G092200_green.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:27%"&gt;
    &lt;a href="https://doi.org/10.1371/journal.pone.0284820" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../nasrin/figs/lung_fpkm_meancorr_eps1.0e+06_r40_g40.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:43%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/myzou_plnt_sci_2500.png"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/mizzou_math_drp.png"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;a href="https://doi.org/10.1101/2022.10.15.512190" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/ipg_network_r.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:25%"&gt;
    &lt;a href="https://doi.org/10.1101/2022.10.15.512190" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../tda/figs/107-tulare.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="list" style="font-size: 14px; text-align: left;"&gt;
    &lt;ul&gt;
      &lt;li&gt;M Bentelspacher, &lt;strong&gt;EJA&lt;/strong&gt;, S Adhikari, J Barros, SY Park (2024) "The early dodder gets the host: Decoding the coiling patterns of Cuscuta campestris with automated image processing". &lt;em&gt;Plant Cell Reports&lt;/em&gt;.&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;EJA&lt;/strong&gt;, MY Quigley, PJ Brown, E Munch, DH Chitwood (2024) "Allometry and volumes in a nutshell: Analyzing walnut morphology using three-dimensional X-ray computed tomography". &lt;em&gt;The Plant Phenome J&lt;/em&gt;.&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;EJA&lt;/strong&gt;, F Nasrin, KM Storey, M Yoshizawa (2023) "Genomics data analysis via spectral shape and topology." &lt;em&gt;PLOS One&lt;/em&gt;&lt;/li&gt;
      &lt;li&gt;SA Cervantes-Pérez &lt;em&gt;et al.&lt;/em&gt; (2024) "Tabula Glycine: The whole-soybean single-cell resolution transcriptome atlas." Submitted.&lt;/li&gt;
      &lt;li&gt;Z Ji, &lt;strong&gt;EJA&lt;/strong&gt;, L Newton, DH Chitwood, AM Thompson (2024) "From hand measurements to high throughput phenotyping: understanding maize canopy structure and predicting yield." Submitted.&lt;/li&gt;
    &lt;/ul&gt;
&lt;/div&gt;

---

background-image: url("../../barley/figs/seed.png")
background-size: 325px
background-position: 99% 99%

class: middle

# Roadmap for today

1. Introduction: Addressing the genotype-phenotype gap

1. The shape of spatial patterns of mRNA

1. The shape of the movement of a vampire plant

1. The shape of omics data analysis

1. The shape of things to come

1. Shaping the next generation of interdisciplinary scientists


---

class: inverse, middle, center

# 1. Introduction

## Shape is data, and data is shape

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:21%"&gt;
    &lt;img src="https://botany.one/wp-content/uploads/2018/08/mcy061.jpg"&gt;&lt;/img&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:31%"&gt;
    &lt;img src="../../mcarto/figs/Infected_Cells_01G164600_green_05G092200_yellow.jpg"&gt;&lt;/img&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:50%"&gt;
    &lt;img src="https://i.kinja-img.com/gawker-media/image/upload/s--usj3b0wY--/c_fit,fl_progressive,q_80,w_636/ve69bswtlq7vqih5qrb1.gif"&gt;&lt;/img&gt;
  &lt;/div&gt;
&lt;/div&gt;

### Visual intuition &amp;harr; Numbers

---

# We use Topological Data Analysis (TDA)!

![](../figs/GLYMA_05G092200_TDA_c00541_transp.gif)

---

# Phenotyping the shape of things to come

&lt;div class="row" style="font-family: 'Yanone Kaffeesatz'; font-size:22px;"&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;text-align: center; font-size:28px"&gt;Phenotyping patterns&lt;/p&gt;
    &lt;img style="padding: 0 30px;" src="../figs/molecular_cartography_2x2.jpg"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 30px;" src="../figs/persistence_images_1x1.svg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;mRNA sub-cellular localization in soybean nodule cells.&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;Phenotyping movement&lt;/p&gt;
    &lt;img style="padding: 0 10px;" src="../../cuscuta/figs/qmlphv.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 35px 30px;" src="../../cuscuta/figs/avg_pl.svg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Tracking and describing &lt;i&gt;Cuscuta campestris&lt;/i&gt; circumnutation&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;Phenotyping data&lt;/p&gt;
    &lt;img style="padding: 0 0px;" src="../../nasrin/figs/fpkm_raw_3.png"&gt;&lt;/img&gt;
    &lt;img style="padding: 30px 0px;" src="../../nasrin/figs/lung_fpkm_meancorr_eps1.0e+06_r40_g40.png"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Reducing &lt;strong&gt;and&lt;/strong&gt; clustering high-dimensional omics data&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

class: inverse, middle, center

# 2. Characterizing mRNA spatial patterns and distributions

## with Topological Data Analysis (TDA)

### In collaboration with Marc Libault

---

# mRNA localization at a sub-cellular level

- Beyond gene expression counts: Spatial segregation and asymmetrical distribution of mRNA across the cytosol in the soybean nodule.

- Molecular Cartography&amp;trade; data provided by the Libault Lab

.pull-left[
![](../figs/Infected_Cells_05G216000_red_05G092200_green.jpg)
Infected soybean nodule cells. Glyma.05G092200 in green. Glyma.05G216000 in red.
]

.pull-right[

**Goals**: "How patterny is a pattern?"

- Quantify the spatial patterns followed by mRNA within individual cells.
- Mathematically model all observed mRNA sub-cellular distributions.
- *Use this mathematical model to differentiate cell types and genotypes.*

**Challenge**

- Develop a mathematical model that works for any cell size, orientation, shape, and dimension.

]

---

# Traditional model: Density of transcripts

![](../figs/sublevel_both_bw3_S25_rho_cyto_GLYMA_05G092200.png)
![](../figs/sublevel_both_bw3_S25_rho_cyto_GLYMA_17G195900.png)

But this characterization discards sub-cellular information!

---

# Same density, different patterns

![](../figs/molecular_cartography_2x4.png)

- 97 genes (including 10 bacterial ones) &amp;rarr; 2 genes
- 2938 cells &amp;rarr; 918 infected ones.

**Subcellular transcript patterns &amp;harr; spatial location of the cell within the nodule**

---


class: inverse, middle, center

# Alternate model: Topological Data Analysis

![](../figs/GLYMA_05G092200_TDA_c00541_transp.gif)

---



# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_00.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_01.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_02.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_03.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_04.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_05.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_06.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_07.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_08.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_09.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_10.png)

---
# TDA: Keep track of blobs and holes

![](../figs/GLYMA_05G092200_TDA_c02352_11.png)


---

# TDA: From patterns to numbers

![](../figs/GLYMA_05G092200_TDA_c02352_13.png)

---

# Do TDA for all cell-gene combinations

&lt;img src="../figs/molecular_cartography_2x4.png" width="500" style="display: block; margin: auto;" /&gt;

&lt;img src="../figs/persistence_images_2x4.png" width="600" style="display: block; margin: auto;" /&gt;

---

background-image: url("../figs/bw25_scale32_-_PI_1_1_1_H1+2_cell_sample.png")
background-size: 620px
background-position: 75% 99%

# PCA on all topological descriptors

&lt;img src="../figs/bw25_both_scale16_-_PI_1_1_1_pca_H1+2_gridded.png" width="350" style="display: block; margin: auto auto auto 0;" /&gt;

---

background-image: url("../figs/bw25_scale32_-_PI_1_1_1_H1+2_kde_sample.png")
background-size: 620px
background-position: 75% 99%

# Show me

&lt;img src="../figs/bw25_both_scale16_-_PI_1_1_1_pca_H1+2_gridded.png" width="350" style="display: block; margin: auto auto auto 0;" /&gt;

---

# Connecting PC 02 to the biological context

![](../figs/sublevel_both_bw3_S25_PCA02_GLYMA_05G092200.png)
![](../figs/sublevel_both_bw3_S25_PCA02_GLYMA_17G195900.png)

- Senescent cells exhibit a distinct transcriptomic spatial pattern compared to the rest of population.
- Loss of mRNA localization may be a lesser known contributor to cell senescence.

---

# We define a morphospace of transcriptomic patterns

![](../figs/bw25_both_scale16_-_PI_1_1_1_pca_H1+2_gridded.png)

# We then work "backward"

---

class: bottom

background-image: url("../figs/scale32_-_PI_1_1_1_H1+2_synthetic_30_clusters.jpg")
background-size: 900px
background-position: 50% 1%

&lt;img src="../figs/scale32_-_PI_1_1_1_H1+2_synthetic_pca_30_clusters.jpg" width="600" style="display: block; margin: auto;" /&gt;

---

class: bottom

background-image: url("../figs/scale32_-_PI_1_1_1_H1+2_synthetic_varclusters.jpg")
background-size: 900px
background-position: 50% 1%

&lt;img src="../figs/scale32_-_PI_1_1_1_H1+2_synthetic_pca_varclusters.jpg" width="600" style="display: block; margin: auto;" /&gt;

---

# Discussion and future directions

**Biologically speaking**

- Senescent cells exhibit a distinct transcriptomic spatial pattern compared to the rest of population.
- Loss of mRNA localization may be a lesser known contributor to cell senescence.
- *How does the morphospace of patterns change if we take into account more genes, more cell types, more tissues, and more mutants?*

**Mathematically speaking**

- Topological Data Analysis offers a robust way to encode the shape of patterns.
- Robust to differences in scale, underlying boundaries, or orientation.
- The framework is open to any number of cells, genes, and dimensions.

&lt;img src="../figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00512.jpg" width="550" style="display: block; margin: auto;" /&gt;

---

class: center, inverse, middle

# 3. Tracking how a vampire plant wiggles

&lt;img src="../../cuscuta/figs/bentelspacher_etal2024.png" width="600" style="display: block; margin: auto;" /&gt;

### In collaboration with Soyon Park

---

background-image: url("../../cuscuta/figs/cuscuta_overview_vogel_etal_2018.webp")
background-size: 350px
background-position: 15% 90%

# _Cuscuta campestris_ : a vegetarian plant

.pull-right[

**Goal**: "How wiggly is a wiggle?"

- Mathematically model how _Cuscuta_ moves under various environmental conditions to ultimately stop it from attaching to crops in the first place.

**Challenges**

- Develop an image-based high-throughput phenotyping algorithm that tracks *Cuscuta* as it coils.
- Develop a mathematical framework that works for any locomotion: circumnutation, twitching, idling, etc.

&lt;p style="font-size: 10px; text-align: left; color: Grey;"&gt;Image Credits: &lt;a href="https://doi.org/10.1038/s41467-018-04344-z"&gt;Vogel &lt;em&gt;et al.&lt;em&gt; (2018)&lt;/a&gt;&lt;/p&gt;

]

---

# Snapshots taken every 96 secs for 24h

&lt;video width="900" controls&gt;
  &lt;source src="../../cuscuta/video/9am_Inc_Rep_3_redone.mp4" type="video/mp4"&gt;
&lt;/video&gt;

- Inoculation at 9am, 12pm, and 4pm
- Experiment setup and data provided by the Park Lab

---

# Putting it all together
  
&lt;video width="900" controls&gt;
  &lt;source src="../../cuscuta/video/4pm_rep7_plant_01.mp4" type="video/mp4"&gt;
&lt;/video&gt;

---

# Automated phenotyping

&lt;img src="../../cuscuta/figs/4pm_rep7_plant_02_posang.png" width="700px" style="display: block; margin: auto;" /&gt;&lt;img src="../../cuscuta/figs/cuscuta_tracking.png" width="700px" style="display: block; margin: auto;" /&gt;

---

# Discussion

![](../../cuscuta/figs/cuscuta_circadian_model.png)

**Biologically speaking**

- Cuscuta can tell time despite lacking photoreceptors.
- It prefers to act in the morning/early afternoon.
- *How does the wiggle vary under other environmental conditions?*

**Computationally speaking**

- Overall, the automated image analysis criteria agreed with the main conclusions drawn from the manual observation criteria. 
- Our pipeline is ready to collect more data

---

background-image: url("../../cuscuta/figs/cuscuta_coords.jpg")
background-size: 450px
background-position: 95% 25%

# Future wiggle room

.pull-left[
**Grant re-submitted to NSF eMB (Emerging Mathematics in Biology) on March 2025**
- Cuscuta locomotion is affected by Volatile Organic Compounds (VOCs).
- Transform a Cuscuta position into a vector of angles.
- Use TDA to characterize all the vectors of all the positions.
]

![](../../cuscuta/figs/tda_pipeline.jpg)

---

class: center, middle inverse

# 4. Phenotyping data itself

## Omics data analysis with topology

&lt;img src="../../nasrin/figs/amezquita_etal_2023.png" width="750" style="display: block; margin: auto;" /&gt;

---

background-image: url("../../nasrin/figs/mapper_vs_tsne_half.png")
background-size: 450px
background-position: 10% 90%


# Setup

- FPKM counts of RNAseq data from human lung tissue &amp;rarr; 19,648 genes
    - 314 healthy samples (GTEx)
    - 500 cancerous samples (TCGA)

- tSNE (or UMAP) separates healthy vs cancerous samples (blue vs red)
   
.pull-right[

**Question**: 

"Is the RNAseq data arranged into a specific shape?"

- Are there subgroups that we are ignoring?
- Can we go from clusters to continua?
- What is the biological characterization of such continua?

]


---

background-image: url("../../tda/figs/mapper_b_00.svg")
background-size: 725px
background-position: 50% 95%

# Mapper 

## Topological summary: exploration and visualization

- We start with **lots** of data points in a **high-dimensional** space.

- We want just a **handful** of points in a **low-dimensional** space that roughly preserve the original **shape**.

---

background-image: url("../../tda/figs/mapper_c_complete.svg")
background-size: 525px
background-position: 50% 99%

# Mapper in a single picture

---

# Mapper and lung cancer data

.pull-left[
![](../../nasrin/figs/t0_1_corr_eps1e+04_r200_g50.png)

![](../../nasrin/figs/lung_t0_meancorr_eps7.0e+02_r40_g30.png)
]

.pull-right[
- Mapper produced mostly strand-like graphs regardless of parameters used

- Healthy subjects tend to stay at the center

- Cancerous samples are distributed at both ends

- Healthy subjects that land in between might be at risk

- **Predictive model**: Take a new patient sample and you can assess its cancer risk based on where they land in this continuum.
]

---

# Biological significance

![](https://journals.plos.org/plosone/article/figure/image?size=large&amp;id=10.1371/journal.pone.0284820.g009)

---

# Discussion and future directions

.pull-left[
- Data visualization to inspire new research.

- Mapper finds novel sub-clusters that reveal important nuances.

- Agnostic to any kind of -omics data

- Mapper remains underused and there is plenty of untapped potential in plant genomics

![](https://www.pnas.org/cms/10.1073/pnas.1102826108/asset/43af8964-ab82-4cbb-bbcb-68f5db437862/assets/graphic/pnas.1102826108fig03.jpeg)
]

.pull-right[

![](../../tda/figs/tissue_leaf_lens.png)

![](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnbt.3854/MediaObjects/41587_2017_Article_BFnbt3854_Fig3_HTML.jpg?as=webp)

]


---

class: inverse, center, middle

# 5. The shape of things to come

## New research frontiers at the intersection of math, data science, and plant biology

### Phenotyping at all scales 

&lt;img src="https://bondlsc.missouri.edu/wp-content/uploads/2015/02/sanborn-620x413.jpg" width="400" style="display: block; margin: auto;" /&gt;

---

background-image: url("../../tutorials/figs/mizzou_math_drp.png")
background-size: 250px
background-position: 99% 1%

# Complex network analysis

.pull-left[

Further analysis of Mapper, gene coexpression, and microbiota interaction networks, and beyond.

![](https://journals.asm.org/cms/10.1128/msystems.01570-24/asset/0d865e60-513c-4bc1-9491-a2919a8e66b6/assets/images/large/msystems.01570-24.f003.jpg)
&lt;p style="font-size: 10px; text-align: left; color: Grey;"&gt;Credits: &lt;a href="https://doi.org/10.1128/msystems.01570-24"&gt;Jiang &lt;em&gt;et al.&lt;/em&gt;(2024)&lt;/a&gt;&lt;/p&gt;

]

.pull-right[

Or even analysis on the collaboration within the IPG

&lt;img src="../../tutorials/figs/ipg_network.jpg" width="300" style="display: block; margin: auto;" /&gt;

]

Math students: Ethan Lenhardt, Sophia Knehans, Roberto Herrera

---

background-image: url("../../tutorials/figs/mizzou_math_drp.png")
background-size: 250px
background-position: 99% 1%

# Spatial topological data analysis

.pull-left[

&lt;img src="../../tda/figs/025-imperial.png" width="200" style="display: block; margin: auto;" /&gt;&lt;img src="../../tda/figs/107-tulare.png" width="200" style="display: block; margin: auto;" /&gt;

&lt;img src="https://www.researchgate.net/publication/362833654/figure/fig2/AS:11431281098979080@1669173266225/Two-persistence-diagrams-for-the-simulation-shown-in-Fig-1-The-blue-crosses-represent.png" width="200" style="display: block; margin: auto;" /&gt;

&lt;p style="font-size: 10px; text-align: center; color: Grey;"&gt;Credits: &lt;a href="https://doi.org/10.1137/19M1241519"&gt;Feng and Porter (2021)&lt;/a&gt;&lt;/p&gt;

]

.pull-right[

Use TDA to analyze geographical patterns across the state.

![](https://i.emlfiles4.com/cmpimg/9/4/0/0/9/1/files/imagecache/1098401/w660_1719770_connectingentrepreneurialcommunitieslogosmall.jpg)

Use TDA to analyze spatial patterns patterns: if you squint enough, a voting district looks pretty much like a plant cell. 

&lt;img src="../../psd/figs/pavement_plasma.jpg" width="200" style="display: block; margin: auto;" /&gt;

]

Math students: Jake Parmentier and Thomas Searcy

---

background-image: url("../../tutorials/figs/mizzou_math_drp.png")
background-size: 250px
background-position: 99% 1%


# Phenotype everywhere!

.pull-left[
![](../../yeast2hybrid/figs/centers_plate_5.jpg)
![](../../yeast2hybrid/figs/p53_P06_contrasting_whiteness_red_1_45_6.png)
![](../../yeast2hybrid/figs/50mM 3AT OEV-control_P01_contrasting_sizes_gray_0_30_4.png)

Image automation (w/ David Mendoza)

]

.pull-right[

&lt;img src="../../psd/figs/MAX_Composite-1.jpg" width="250" style="display: block; margin: auto;" /&gt;
Spatial transcriptomics (w/ Jie Zhu)

&lt;img src="../../root_necrosis/figs/geodesic_comparison_-_222M_side1_030117006.png" width="250" style="display: block; margin: auto;" /&gt;

2D root image analysis (w/ Miranda Haus (MSU))

]

---

background-image: url("https://plantsandpython.github.io/PlantsAndPython/_images/plants_python_logo.jpg")
background-size: 180px
background-position: 99% 1%

class: inverse, center, middle

# 6. Shaping the next generation of interdisciplinary scientists

## Large amounts of data require large amounts of people

## PLNT_SCI 2500: Data Science for Life Sciences I

![](../../tutorials/figs/myzou_plnt_sci_2500.png)

---

## PLNT_SCI 2500: Python taught for life sciences

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-10_In-Class_NumPyDataAnalysis2-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---


## PLNT_SCI 2500: With data from DPST faculty!

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-13_In-Class_Regression-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---


## PLNT_SCI 2500: Active learning, example driven

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-20_In-Class_AdvancedPlotting-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---

background-image: url("https://www.biorxiv.org/content/biorxiv/early/2022/09/09/2022.09.07.506951/F1.large.jpg?width=800&amp;height=600&amp;carousel=1")
background-size: 325px
background-position: 1% 60%

# Discussion and future goals

**PLNT_SCI 2500: Data Science for Life Sciences I will be the first course for the emerging Data Science for Life Sciences Certificate**

.right-column[

- Incorporate examples from outside Plant Science to appeal to more students

- Data Science for Life Sciences II will build on top more Data Science topics:
    - Supervised and unsupervised classification
    - Clustering algorithms
    - Non-linear regressions
    - Network analyses
    - Statistical paradoxes to be wary of
    
**Personal Goal: Make the teaching of data science in life sciences a scientific endeavor in itself**

- Collaborate with colleagues from the College of Education.
- Motivate students to tackle a single research problem as a unit.

]

---

background-image: url("https://upload.wikimedia.org/wikipedia/commons/4/4a/University_of_Missouri_logo.svg")
background-size: 60px
background-position: 99% 1%

class: inverse

## Thank you!

.pull-left[

**mRNA sub-cellular localization**

- Sutton Tennant
- Sandra Thibivillers
- Sai Subhash
- Benjamin Smith
- Samik Bhattacharya
- Jasper Kläver
- Marc Libault

**Cuscuta circadian rhythm and locomotion**

- Max Bentelspacher
- Supral Adhikari
- Jaime Barros-Rios
- Joseph Lynch
- So-Yon Park

**Mapper for omics analysis**

- Farzana Nasrain
- Katie Storey
- Masato Yoshizawa

]

.pull-right[

**Collaboration of the IPG network**

- Ethan Lenhardt
- Sophia Knehans
- Roberto Herrera
- David Braun

**Data Science for Life Sciences I**

- Kent Shannon
- Andrew Scaboo
- Jianfeng Zhou
- Debbie Finke

**Other ongoing projects**

- Leyre Urmeneta
- Laura Martins
- Mather Khan
- David Mendoza-Cozatl
- Miranda Haus
- Jie Zhu

**More details**

&lt;p style="font-size: 20px; text-align: center; color: Blue;"&gt;ejamezquita.github.io/&lt;/p&gt;
&lt;p style="font-size: 20px; text-align: center; color: Blue;"&gt;eah4d@missouri.edu&lt;/p&gt;

]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="../../js/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
