<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>The mathematical shape of plants</title>
    <meta charset="utf-8" />
    <meta name="author" content="Erik Amézquita — Division of Plant Science &amp; Technology Department of Mathematics University of Missouri —" />
    <meta name="date" content="2026-01-13" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../css/mu.css" type="text/css" />
    <link rel="stylesheet" href="../../css/gallery.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# The mathematical shape of plants
]
.subtitle[
## Topological Data Analysis and plant science
]
.author[
### Erik Amézquita<br>—<br>Division of Plant Science &amp; Technology<br>Department of Mathematics<br>University of Missouri<br>—
]
.date[
### 2026-01-13
]

---




background-image: url("../../demat/figs/fam9_3.png")
background-size: 100px
background-position: 98% 2%

# About me: From MX to MI to MO at MU

## I work across multiple disciplines and countries

.left-column[
![](../../img/dvdy216-toc-0001-m.jpg)
]

.right-column[
- 2013 - 2018 : Licenciatura (Bachelor): Mathematics @ Universidad de Guanajuato and CIMAT. 
- 2018 - 2023 : PhD: Computational Mathematics, Science &amp;amp; Engineering @ Michigan State University 
- 2023 - 2025 : ~~PFFFD~~ ~~PFFIE~~ PFF Postdoctoral Fellow: Plant Science / Mathematics @ University of Missouri (MU).
- 2025 - present: Assistant Professor: Plant Science / Math @ MU. 
]

---

background-image: url("../../demat/figs/ecg_results.png")
background-size: 700px
background-position: 50% 80%

# College &amp;rarr; TDA &amp;oplus; Archaeology

Topological Data Analysis for pre-Columbian mask classification

--

.pull-left[

&lt;img src="../../demat/figs/barrett_poster-1.jpg" alt="" width="310" style="display: block; margin: auto;" /&gt;
]

--

.pull-right[
![](https://cdn.pixabay.com/photo/2023/09/29/21/41/cancelled-8284622_960_720.png)


``` r
*- Lessons learned for better 
* future interdisciplinary 
* collaborations
  
*- Always include the other
* discipline experts at
* every stage of the analysis

*- Learn the basics of other
* people disciplines
```
]

---

# TDA &amp;oplus; Archaeology: retold

.pull-left[
![](../../demat/figs/mezcala_masks.png)

![](../../demat/figs/hairdo.png)
&lt;p style="font-size: 10px; text-align: center; color: Grey;"&gt;Credits: &lt;a href="http://mediateca.inah.gob.mx/repositorio/islandora/object/tesis%3A4497"&gt;Olmedo Vera and Gonz&amp;aacute;lez (1986)&lt;/a&gt;&lt;/p&gt;
]

.pull-right[
![](../../demat/figs/olmedoVera_gonzalez_1986.png)
]

---

class: inverse, center, middle

# A mathematician by training, a data scientist by trade, and a plant biologist by ~~collaboration~~ osmosis

&lt;img src="../../biology/figs/osmosis-in-plant-cell.png" alt="" width="400" style="display: block; margin: auto;" /&gt;

---

background-image: url("../../arabidopsis/figs/PFig1.png")
background-size: 900px
background-position: 50% 70%

# Shape quantification &amp;oplus; Plant Science

- Quantifying the intuition
- Intuifying the quantities

---

# Come for the math, stay for the plants

&lt;div class="row" style="font-family: 'Yanone Kaffeesatz'; font-size:22px;"&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;text-align: center; font-size:28px"&gt;The shape of adaptability&lt;/p&gt;
    &lt;img style="padding: 0 70px;" src="../../barley/figs/S017_L3_1.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 55px;" src="../../barley/figs/ecc_X.gif"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Topological Data Analysis&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Euler Characteristic Transform&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;The shape of development&lt;/p&gt;
    &lt;img style="padding: 0 60px;" src="../../citrus/figs/SR01_L01_black_exocarp.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 27px;" src="../../citrus/figs/SW03_CRC1241-B_12B-4-3_L00_lambproj_v.jpg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Ellipsoidal modeling&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Directional statistics&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;The shape of domestication&lt;/p&gt;
    &lt;img style="padding: 0 35px;" src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 10px;" src="../../walnuts/figs/kernel_volume_vs_nut_volume.png"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Allometry of multiple tissues&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Convexity indices&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

# Phenotyping the shape of things to come

&lt;div class="row" style="font-family: 'Yanone Kaffeesatz'; font-size:22px;"&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;text-align: center; font-size:28px"&gt;Phenotyping patterns&lt;/p&gt;
    &lt;img style="padding: 0 30px;" src="../../mcarto/figs/molecular_cartography_2x2.jpg"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 30px;" src="../../mcarto/figs/persistence_images_1x1.svg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;mRNA sub-cellular localization in soybean nodule cells.&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;Phenotyping movement&lt;/p&gt;
    &lt;img style="padding: 0 10px;" src="../../cuscuta/figs/qmlphv.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 35px 30px;" src="../../cuscuta/figs/avg_pl.svg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Tracking and describing &lt;i&gt;Cuscuta campestris&lt;/i&gt; circumnutation&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;Phenotyping data&lt;/p&gt;
    &lt;img style="padding: 0 0px;" src="../../nasrin/figs/fpkm_raw_3.png"&gt;&lt;/img&gt;
    &lt;img style="padding: 30px 0px;" src="../../nasrin/figs/lung_fpkm_meancorr_eps1.0e+06_r40_g40.png"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Reducing &lt;strong&gt;and&lt;/strong&gt; clustering high-dimensional omics data&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---


class: inverse, middle, center

# The culture shock of switching disciplines has been greater than that of switching countries

&lt;img src="../../img/homotopy_botany.png" alt="" width="500" style="display: block; margin: auto;" /&gt;

# But it is fun

---

background-image: url("../../barley/figs/seed.png")
background-size: 325px
background-position: 99% 99%

class: middle

# Roadmap for today

1. Phenotyping shapes: barley seeds
   - Euler Characteristic Transform

1. Phenotyping patterns: mRNA localization in soybean
   - Sublevel set persistence
   - Persistence Images

1. Phenotyping data: lung tissue RNAseq 
   - Mapper
   
1. Current plant data and TDA opportunities

---

class: inverse, middle, center

# Phenotyping shapes

## How _shape-y_ &amp;nbsp; is a shape?

&lt;img src="../../barley/figs/amezquita_etal_2021.png" alt="" width="500" style="display: block; margin: auto;" /&gt;

---

# Traditional Morphometry

&lt;img src="https://nph.onlinelibrary.wiley.com/cms/asset/e0e8b362-efee-410a-a293-947c2d57acb9/nph16286-fig-0003-m.jpg" alt="" width="650" style="display: block; margin: auto;" /&gt;
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt; Credits: &lt;a href="https://doi.org/10.1111/nph.16286"&gt;Gupta &lt;em&gt;et al.&lt;/em&gt; (2019)&lt;/a&gt;&lt;/p&gt;

---

# Modern Morphometric Methods might not be enough

.pull-left[
Elliptical Fourier Descriptor (EFD)
![](../../barley/figs/chitwood_sinha_2016_01_02.jpg)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt; Credits: &lt;a href="https://doi.org/10.1016/j.cub.2016.02.033"&gt;Chitwood and Sinha (2016)&lt;/a&gt;&lt;/p&gt;
]

.pull-right[
Landmark-based morphometrics (GMM)
![](../../barley/figs/chitwood_2021_fig-004_1.jpg)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt; Credits: &lt;a href=" https://doi.org/10.1002/ppp3.10157"&gt;Chitwood  (2020)&lt;/a&gt;&lt;/p&gt;
]

---

class: inverse

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:44%"&gt;
    &lt;a href="https://cereal.interreg-npa.eu/subsites/CEREAL/Barley_cultivation-Norway_Presentation_2018.pdf" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../barley/figs/barley_norway.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://www.resilience.org/stories/2020-03-09/the-last-crop-before-the-desert/" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../barley/figs/barley_morocco.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://www.doi.org/10.1007/978-1-4419-0465-2_2168" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../barley/figs/barley_world.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:44%"&gt;
    &lt;a href="https://www.bloomberg.com/news/articles/2020-09-13/iraq-to-offer-first-ever-barley-exports-as-rains-yield-surplus" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../barley/figs/barley_iraq.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://www.tibettravel.org/tibetan-culture/highland-barley.html" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../barley/figs/barley_seed_tibet.jpg"&gt;&lt;/a&gt;
    &lt;div class="row"&gt;
      &lt;div class="column" style="max-width:46%"&gt;
        &lt;img src="../../barley/figs/S019_L0_1.gif"&gt;
      &lt;/div&gt;
      &lt;div class="column" style="max-width:55%"&gt;
        &lt;img src="../../barley/figs/S017_L0_seed_10_0.gif"&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:8%; font-size: 15px;"&gt;
    &lt;p style="text-align: center; font-size: 30px; line-height: 1em;"&gt; &lt;strong&gt; Barley from across the world &lt;/strong&gt;&lt;/p&gt;
    &lt;p&gt;28 collected barley varieties&lt;/p&gt;
    &lt;p&gt;Brought to California in 1929&lt;/p&gt;
    &lt;p&gt;Artificial evolution experiment for 58 generations&lt;/p&gt;
    &lt;p&gt;975 panicles scanned&lt;/p&gt;
    &lt;p&gt;38,000 seeds isolated&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;


---

## The Euler Characteristic Curve (ECC)

&lt;img src="../../barley/figs/ecc_ver2.gif" alt="" width="650" style="display: block; margin: auto;" /&gt;

## The Euler Characteristic Transform (ECT)

&lt;img src="../../barley/figs/ect_000.png" alt="" width="650" style="display: block; margin: auto;" /&gt;

---

## The Euler Characteristic Curve (ECC)

&lt;img src="../../barley/figs/ecc_Z_32.png" alt="" width="650" style="display: block; margin: auto;" /&gt;

## The Euler Characteristic Transform (ECT)

&lt;img src="../../barley/figs/ect_ver2.gif" alt="" width="650" style="display: block; margin: auto;" /&gt;

---

# Quantify the shape of barley

**Goal:** Classify the **28** founding barley varieties solely by grain morphology information.

&lt;style type="text/css"&gt;
.tg  {border-collapse:collapse;border-color:#93a1a1;border-spacing:0;margin:0px auto;}
.tg td{background-color:#fdf6e3;border-bottom-width:1px;border-color:#93a1a1;border-style:solid;border-top-width:1px;
  border-width:0px;color:#002b36;font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 5px;
  word-break:normal;}
.tg th{background-color:#657b83;border-bottom-width:1px;border-color:#93a1a1;border-style:solid;border-top-width:1px;
  border-width:0px;color:#fdf6e3;font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;
  padding:10px 5px;word-break:normal;}
.tg .tg-2bhk{background-color:#eee8d5;border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-0pky{border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-gyvr{background-color:#eee8d5;border-color:inherit;font-size:100%;text-align:left;vertical-align:top}
&lt;/style&gt;
&lt;table class="tg"&gt;
&lt;thead&gt;
  &lt;tr&gt;
    &lt;th class="tg-0pky"&gt;Shape descriptors&lt;/th&gt;
    &lt;th class="tg-0pky"&gt;No. of descriptors&lt;/th&gt;
    &lt;th class="tg-0pky"&gt;F1&lt;/th&gt;
  &lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
    &lt;td class="tg-2bhk"&gt;Traditional&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;11&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;0.55 &amp;plusmn; 0.019&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td class="tg-0pky"&gt;Topological&lt;/td&gt;
    &lt;td class="tg-0pky"&gt;12&lt;/td&gt;
    &lt;td class="tg-0pky"&gt;0.74 &amp;plusmn; 0.016&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td class="tg-2bhk"&gt;Combined&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;23&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;&lt;strong&gt;0.86 &amp;plusmn; 0.010&lt;/strong&gt;&lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

### What does topology actually measure?

.pull-left[
&lt;img src="../../barley/figs/discerning_directions.png" alt="" width="225" style="display: block; margin: auto;" /&gt;
]

.pull-right[
![](../../barley/figs/arrow_seed_09_0.gif)
]

---

# Into semi-supervised territory

- Train an SVM with 100% of the founders `\((F_0)\)`

- Classify the progeny `\(F_{58}\)`: DNA (genotype) &amp;harr; shape (phenotype) enrichment

.pull-left[
![](../../barley/figs/histogram_combined_gen0_d158_T16_topounscaled_vert.svg)
]

.pull-right[
![](../../barley/figs/histogram_combined_gen7_d158_T16_topounscaled_vert.svg)
]

---

background-image: url("../figs/amezquita_etal_2021.png")
background-size: 300px
background-position: 98% 2%


# Summary

.left-column[
&lt;img src="../../barley/figs/S017_L3_1.gif" alt="" width="150" style="display: block; margin: auto;" /&gt;&lt;img src="../../barley/figs/ecc_X.gif" alt="" width="150" style="display: block; margin: auto;" /&gt;
]

.right-column[
- **TDA** captures **nuanced** morphological information that is not obvious to the naked eye.

- We get the best results by **combining** both traditional and topological morphological descriptors.

- We can tell **similar** population genetic dynamics by using combined **morphological** descriptors.

- Seed phenotyping can be **cheaper** than genotyping.
]


---

# Color me rooted and phenotyped!

.pull-left[
![](../../yeast2hybrid/figs/plate_02_p53_plate_02_raw.pdf.png)
![](../../yeast2hybrid/figs/p53_plate_pipeline_ver04.jpg)

Persistent intensity peak color
]

.pull-right[

![](../../root_necrosis/figs/geodesic_comparison_-_222M_side1_030117006.png)

![](../../root_necrosis/figs/1-CAL-1-M212_-_geodesic_distance_transform.png)

Soybean root phenotyping across years

]



---

class: inverse, middle, center

# Phenotyping patterns

## How *pattern-y* is a pattern?

(w/ 
    Sutton Tennant,
    Sandra Thibivillers,
    Sai Subhash,
    Benjamin Smith,
    Samik Bhattacharya,
    Jasper Kläver,
    Marc Libault
)

---

# The Central Dogma and Gene Expression

.pull-left[

![](https://images.squarespace-cdn.com/content/v1/52668d02e4b0f593739ec2b6/1533240020250-N1ISWHD91G3VPFJAN7YP/1.png)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt;Credits: &lt;a href="https://doi.org/10.1242/bio.017178"&gt;Xu &lt;em&gt;et al.&lt;/em&gt; (2016)&lt;/a&gt;&lt;/p&gt;
]

.pull-right[

- Every cell in an organism has the same* DNA.
- But cells will only express a subset of genes depending on their type. 
- By knowing which genes are over/under-expressed, we can gather what makes each cell type unique and what are their exact functions.

![](https://www.researchgate.net/publication/303559263/figure/fig3/AS:366370148634637@1464360858751/A-heat-map-showing-SmARF-gene-expression-patterns-in-different-organs-The-red-color.png)
]

---

# mRNA localization at a sub-cellular level

- Beyond gene expression counts: Spatial segregation and asymmetrical distribution of mRNA across the cytosol in the soybean nodule.

- Molecular Cartography&amp;trade; data provided by the Libault Lab

.pull-left[
![](../../mcarto/figs/Infected_Cells_05G216000_red_05G092200_green.jpg)
Infected soybean nodule cells. Glyma.05G092200 in green. Glyma.05G216000 in red.
]

.pull-right[

**Goals**: "How patterny is a pattern?"

- Quantify the spatial patterns followed by mRNA within individual cells.
- Mathematically model all observed mRNA sub-cellular distributions.
- *Use this mathematical model to differentiate cell types and genotypes.*

**Challenge**

- Develop a mathematical model that works for any cell size, orientation, shape, and dimension.

]

---

# Same expression levels, different patterns

![](../../mcarto/figs/molecular_cartography_2x4.png)

**Subcellular transcript patterns &amp;harr; spatial location of the cell within the nodule**

---

# How *pattern-y* is a pattern?

![](../../mcarto/figs/cell_sample.jpg)

Patterns are subject to a large variety of cell shapes, sizes, orientations, and mRNA quantities.
- 97 genes (including 10 bacterial ones) &amp;rarr; 6 genes
- 2938 cells &amp;rarr; 918 infected ones.


---

background-image: url("../../mcarto/figs/GLYMA_05G092200_TDA_c00541_transp.gif")
background-size: 800px
background-position: 50% 85%

# KDEs &amp;oplus; Sublevel sets &amp;oplus; Pers. images

---

# Focus on `\(H_1\)` and `\(H_2\)`

&lt;img src="../../mcarto/figs/molecular_cartography_2x4.png" alt="" width="500" style="display: block; margin: auto;" /&gt;

&lt;img src="../../mcarto/figs/persistence_images_2x4.png" alt="" width="600" style="display: block; margin: auto;" /&gt;

---

background-image: url("../../mcarto/figs/bw25_scale32_-_PI_1_1_1_H1+2_cell_sample.png")
background-size: 620px
background-position: 75% 99%

# PCA on all topological descriptors

&lt;img src="../../mcarto/figs/bw25_both_scale16_-_PI_1_1_1_pca_H1+2_gridded.png" alt="" width="350" style="display: block; margin: auto auto auto 0;" /&gt;

---

background-image: url("../../mcarto/figs/bw25_scale32_-_PI_1_1_1_H1+2_kde_sample.png")
background-size: 620px
background-position: 99% 50%

.left-column[

# PC 1
- Related to the number of distinct hotspots
- Correlated to transcript number and cell size

# PC 2
- Related to the heterogeneity of hotspots
- Correlated to transcript density

]

---

# Connecting PC 02 to the biological context

![](../../mcarto/figs/GLYMA_17G195900_bw15_PI_scale24_fulleccentricity_correlations.png)

- Senescent cells exhibit a distinct transcriptomic spatial pattern compared to the rest of population.
- Loss of mRNA localization may be a lesser known contributor to cell senescence.

---

# We define a morphospace of transcriptomic patterns

![](../../mcarto/figs/bw25_both_scale16_-_PI_1_1_1_pca_H1+2_gridded.png)

# We then work "backward"

---

class: bottom

background-image: url("../../mcarto/figs/scale32_-_PI_1_1_1_H1+2_synthetic_30_clusters.jpg")
background-size: 900px
background-position: 50% 1%

&lt;img src="../../mcarto/figs/scale32_-_PI_1_1_1_H1+2_synthetic_pca_30_clusters.jpg" alt="" width="600" style="display: block; margin: auto;" /&gt;

---

class: bottom

background-image: url("../../mcarto/figs/scale32_-_PI_1_1_1_H1+2_synthetic_varclusters.jpg")
background-size: 900px
background-position: 50% 1%

&lt;img src="../../mcarto/figs/scale32_-_PI_1_1_1_H1+2_synthetic_pca_varclusters.jpg" alt="" width="600" style="display: block; margin: auto;" /&gt;

---

# Summary

**Biologically speaking**

- Senescent cells exhibit a distinct transcriptomic spatial pattern compared to the rest of population.
- Loss of mRNA localization may be a lesser known contributor to cell senescence.
- *How does the morphospace of patterns change if we take into account more genes, more cell types, more tissues, and more mutants?*

**Mathematically speaking**

- Topological Data Analysis offers a robust way to encode the shape of patterns.
- Robust to differences in scale, underlying boundaries, or orientation.
- The framework is open to any number of cells, genes, and dimensions.

&lt;img src="../../mcarto/figs/D2_GLYMA_05G092200_z_kde_pd_suplevel_by_both_00512.jpg" alt="" width="550" style="display: block; margin: auto;" /&gt;

---


# Gerrymandering tears and battlefields

.pull-left[

&lt;img src="../../tda/figs/025-imperial.png" alt="" width="220" style="display: block; margin: auto;" /&gt;&lt;img src="../../tda/figs/107-tulare.png" alt="" width="220" style="display: block; margin: auto;" /&gt;&lt;img src="../../psd/figs/pavement_plasma.jpg" alt="" width="220" style="display: block; margin: auto;" /&gt;


Voting districts &amp;asymp; pavement cells. 

]

.pull-right[
![](../../psd/figs/MAX_Composite-1.jpg)
![](../../psd/figs/Fig_1D_Raw_normalizing.png)

Spatial transcriptomics of a battlefield

]


---

class: inverse, middle, center

# Phenotyping data

## The shape of omics data and molecular biology

&lt;img src="../../nasrin/figs/amezquita_etal_2023.png" alt="" width="750" style="display: block; margin: auto;" /&gt;


---

# The Central Dogma and Gene Expression

.pull-left[

![](https://images.squarespace-cdn.com/content/v1/52668d02e4b0f593739ec2b6/1533240020250-N1ISWHD91G3VPFJAN7YP/1.png)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt;Credits: &lt;a href="https://doi.org/10.1242/bio.017178"&gt;Xu &lt;em&gt;et al.&lt;/em&gt; (2016)&lt;/a&gt;&lt;/p&gt;
]

.pull-right[

- Every cell in an organism has the same* DNA.
- But cells will only express a subset of genes depending on their type. 
- By knowing which genes are over/under-expressed, we can gather what makes each cell type unique and what are their exact functions.

![](https://www.researchgate.net/publication/303559263/figure/fig3/AS:366370148634637@1464360858751/A-heat-map-showing-SmARF-gene-expression-patterns-in-different-organs-The-red-color.png)
]

---

background-image: url("../../nasrin/figs/mapper_vs_tsne_half.png")
background-size: 450px
background-position: 10% 90%


# Problem: Data is very high-dimensional

- FPKM counts of RNAseq data from human lung tissue &amp;rarr; 19,648 genes
    - 314 healthy samples (GTEx)
    - 500 cancerous samples (TCGA)

- tSNE (or UMAP) separates healthy vs cancerous samples (blue vs red)
   
.pull-right[

**Question**: 

"Is the RNAseq data arranged into a specific shape?"

- Are there subgroups that we are ignoring?
- Can we go from clusters to continua?
- What is the biological characterization of such continua?

]


---

background-image: url("../../tda/figs/mapper_b_00.svg")
background-size: 725px
background-position: 50% 95%

# Mapper 

## Topological summary: exploration and visualization

- We start with **lots** of data points in a **high-dimensional** space.

- We want just a **handful** of points in a **low-dimensional** space that roughly preserve the original **shape**.

---

background-image: url("../../tda/figs/mapper_c_complete.svg")
background-size: 525px
background-position: 50% 99%

# Mapper in a single picture

---

# Mapper and lung cancer data

.pull-left[
![](../../nasrin/figs/t0_1_corr_eps1e+04_r200_g50.png)

![](../../nasrin/figs/lung_t0_meancorr_eps7.0e+02_r40_g30.png)
]

.pull-right[
- Mapper produced mostly strand-like graphs regardless of parameters used

- Healthy subjects tend to stay at the center

- Cancerous samples are distributed at both ends

- Healthy subjects that land in between might be at risk

- **Predictive model**: Take a new patient sample and you can assess its cancer risk based on where they land in this continuum.
]

---

# Biological significance

![](https://journals.plos.org/plosone/article/figure/image?size=large&amp;id=10.1371/journal.pone.0284820.g009)

---

# Summary

.pull-left[
- Data visualization to inspire new research.

- Mapper finds novel sub-clusters that reveal important nuances.

- Agnostic to any kind of -omics data

- Mapper remains underused and there is plenty of untapped potential in plant genomics

![](https://www.pnas.org/cms/10.1073/pnas.1102826108/asset/43af8964-ab82-4cbb-bbcb-68f5db437862/assets/graphic/pnas.1102826108fig03.jpeg)
]

.pull-right[

![](../../tda/figs/tissue_leaf_lens.png)

![](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnbt.3854/MediaObjects/41587_2017_Article_BFnbt3854_Fig3_HTML.jpg?as=webp)

]


---


background-image: url("../../nasrin/figs/rice_mapper_2025-11-20.png")
background-size: 905px
background-position: 50% 90%

# Leading the Resistance with mapper

- RNASeq data from rice samples: control and bacterial infection
- Look for differentially expressed genes to understand disease resistance


---

background-image: url("../../tutorials/figs/knehansIPG.jpg")
background-size: 470px
background-position: 0% 85%

# Since we're talking about networks

.pull-right[
**Studying the people that study plants**

- Academic social network analysis inspired by the IPG (Interdisciplinary Plant Group) at the University of Missouri

- Expanded data collection for all 51 land-grant institutions

- Weighted graph:
    - Plant faculty: nodes
    - No. of papers together: edge
    
- Does collaboration/interdisciplinarity lead to better academic outcomes?

- How to quantify collaboration/interdisciplinarity?

W/ Roberto Herrera, Sophia Knehans, David Braun
]


---

background-image: url("https://plantsandpython.github.io/PlantsAndPython/_images/plants_python_logo.jpg")
background-size: 180px
background-position: 99% 1%

class: inverse, center, middle

# Shaping the next generation of interdisciplinary scientists

## Large amounts of data require large amounts of people

(Ask me later about my undergrad course on Data Science to Life Sciences)

---

## Pers. Diagrams &amp;oplus; distances &amp;oplus; MDS &amp;oplus; PCA

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/slides/tda_distances_pca_pipeline.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---

## Mapper without the need to code

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/slides/inital_mapper_pipeline.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;


---

background-image: url("https://upload.wikimedia.org/wikipedia/commons/4/4a/University_of_Missouri_logo.svg")
background-size: 60px
background-position: 99% 1%

class: inverse

## Thank you!

.pull-left[

**Barley seeds &amp;oplus; ECT**

- Liz Munch
- Dan Chitwood
- Michelle Quigley
- Tim Ophelders
- Jacob Landis
- Dan Koenig

**mRNA sub-cellular localization**

- Sutton Tennant
- Sandra Thibivillers
- Sai Subhash
- Benjamin Smith
- Samik Bhattacharya
- Jasper Kläver
- Marc Libault

**Mapper for lung cancer**

- Farzana Nasrain
- Katie Storey
- Masato Yoshizawa


]

.pull-right[

**Collaboration of the IPG network**

- Ethan Lenhardt
- Sophia Knehans
- Roberto Herrera
- David Braun

**Other ongoing TDA projects**

- Laura Martins
- Mather Khan
- David Mendoza-Cozatl
- Jie Zhu
- Felix Fritschi
- Jose Costa Netto
- Tim Duff
- Olivia Fisk
- Gloria Asare
- Ajay Gupta
- Bing Yang
- Tyler Mitts
- Colin Nichols


**More details**

&lt;p style="font-size: 20px; text-align: center; color: Blue;"&gt;ejamezquita.github.io/&lt;/p&gt;
&lt;p style="font-size: 20px; text-align: center; color: Blue;"&gt;eah4d@missouri.edu&lt;/p&gt;

]


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="../../js/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
  "highlightStyle": "github",
  "highlightLines": true,
  "countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
