<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>The shape of things to come</title>
    <meta charset="utf-8" />
    <meta name="author" content="Erik Amézquita   —   Computational Mathematics, Science, and Engineering   Michigan State University   —" />
    <meta name="date" content="2023-04-11" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../css/msu.css" type="text/css" />
    <link rel="stylesheet" href="../../css/gallery.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# The shape of things to come
]
.subtitle[
## Exploring the mathematical shape of plants
]
.author[
### <strong>Erik Amézquita</strong> <br> — <br> Computational Mathematics, Science, and Engineering <br> Michigan State University <br> —
]
.date[
### 2023-04-11
]

---




background-image: url("../../img/endlessforms.png")
background-size: 150px
background-position: 89% 7%

class: inverse

# Plant morphology

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:50%"&gt;
    &lt;iframe width="375" height="210" src="https://www.youtube-nocookie.com/embed/oM9kAq0PBvw?controls=0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;&lt;/iframe&gt;
    &lt;iframe width="375" height="210" src="https://www.youtube-nocookie.com/embed/V39K58evWlU?controls=0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;&lt;/iframe&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:50%"&gt;
    &lt;iframe width="375" height="210" src="https://www.youtube-nocookie.com/embed/efF5PSvFQ2A?controls=0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;&lt;/iframe&gt;
    &lt;iframe width="375" height="210" src="https://www.youtube-nocookie.com/embed/qkOjHHuoUhA?controls=0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;&lt;/iframe&gt;
  &lt;/div&gt;
&lt;/div&gt;
&lt;p style="font-size: 24px; text-align: right; font-family: 'Yanone Kaffeesatz'"&gt;Check out more 3D X-ray CT scans at &lt;a href="https://www.youtube.com/@endlessforms6756"&gt;youtube.com/@endlessforms6756&lt;/a&gt;&lt;/p&gt;

---

background-image: url("../../cmse/figs/cmse logo file-01.svg")
background-size: 150px
background-position: 1% 50%

class: center

# My research: Crossing and merging bridges

&lt;div class="row"&gt;
  &lt;div class="column" style="width:15%"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:15%"&gt;
    &lt;a href="https://doi.org/10.1093/insilicoplants/diab033" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/S017_L3_1.gif"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:18%"&gt;
    &lt;a href="https://doi.org/10.1002/ppp3.10333" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../citrus/figs/SR01_L01_black_exocarp.gif"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:25%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../arabidopsis/figs/Day7_110921_U112-3_pot0_leaf_X.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:25%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif"&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row"&gt;
  &lt;div class="column" style="width:15%"&gt;
  &lt;p style="font-size:5px"&gt;&amp;mdash;&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:28%"&gt;
    &lt;a href="https://plantsandpython.github.io/PlantsAndPython/00_Opening_page.html" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/plants_and_python_group_photo.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:50%"&gt;
    &lt;a href="https://doi.org/10.1073/pnas.2217564120" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../journal_diversity/figs/frequencies_globe_inv.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="list" style="font-size: 10px; text-align: left;"&gt;
    &lt;ul&gt;
      &lt;li&gt;Marks &lt;em&gt;et al.&lt;/em&gt; (2023) A critical analysis of plant science literature reveals ongoing inequities. DOI: &lt;a href="https://doi.org/10.1073/pnas.2217564120" target="_blank"&gt;10.1101/2022.10.15.512190&lt;/a&gt;&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;A&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2022) The shape of aroma: measuring and modeling citrus oil gland distribution. DOI: &lt;a href="https://doi.org/10.1002/ppp3.10333" target="_blank"&gt;10.1002/ppp3.10333&lt;/a&gt;&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;A&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2022) Genomics data analysis via spectral shape and topology. DOI: &lt;a href="https://doi.org/10.48550/arXiv.2211.00938" target="_blank"&gt;10.48550/arXiv.2211.00938&lt;/a&gt;&lt;/li&gt;
      &lt;li&gt;VanBuren&lt;em&gt;et al.&lt;/em&gt; (2022) Teaching tools in plant biology. Plants and Python, coding from scratch in the plant sciences. DOI: &lt;a href="https://doi.org/10.1093/plcell/koac187" target="_blank"&gt;10.1093/plcell/koac187&lt;/a&gt;.&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;A&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2021) Measuring hidden phenotype: Quantifying the shape of barley seeds using the Euler Characteristic Transform. DOI: &lt;a href="https://doi.org/10.1093/insilicoplants/diab033" target="_blank"&gt;10.1093/insilicoplants/diab033&lt;/a&gt;&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;A&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2020) The shape of things to come: Topological data analysis and biology, from molecules to organisms. DOI: &lt;a href="https://doi.org/10.1002/dvdy.175" target="_blank"&gt;10.1002/dvdy.175&lt;/a&gt;&lt;/li&gt;
    &lt;/ul&gt;
&lt;/div&gt;

---

# The shape of things to come

&lt;div class="row" style="font-family: 'Yanone Kaffeesatz'; font-size:22px;"&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;text-align: center; font-size:28px"&gt;The shape of adaptability&lt;/p&gt;
    &lt;img style="padding: 0 70px;" src="../figs/S017_L3_1.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 55px;" src="../figs/ecc_X.gif"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Topological Data Analysis&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Euler Characteristic Transform&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;The shape of development&lt;/p&gt;
    &lt;img style="padding: 0 60px;" src="../../citrus/figs/SR01_L01_black_exocarp.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 27px;" src="../../citrus/figs/SW03_CRC1241-B_12B-4-3_L00_lambproj_v.jpg"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Ellipsoidal modeling&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Directional statistics&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%"&gt;
    &lt;p style="line-height:0;font-size:28px;text-align: center;"&gt;The shape of domestication&lt;/p&gt;
    &lt;img style="padding: 0 35px;" src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif"&gt;&lt;/img&gt;
    &lt;img style="padding: 0 10px;" src="../../walnuts/figs/kernel_volume_vs_nut_volume.png"&gt;&lt;/img&gt;
    &lt;p style="text-align: center;"&gt;Allometry of multiple tissues&lt;/p&gt;
    &lt;p style="text-align: center;"&gt;Convexity indices&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

class: inverse, middle, center

# 1. Quantifying the barley morphology

## Brewing with the Euler Characteristic Transform

&lt;img src="../figs/amezquita_etal_2021.png" width="400" style="display: block; margin: auto;" /&gt;

---

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:44%"&gt;
    &lt;a href="https://kizilvest.ru/20150827-v-kizilskom-rajone-nachalas-uborochnaya-strada/" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_kizilskoye.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://ipad.fas.usda.gov/highlights/2008/11/eth_25nov2008/" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_ethiopia.gif"&gt;&lt;/a&gt;
    &lt;a href="https://www.doi.org/10.1007/978-1-4419-0465-2_2168" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_historical_expansion.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:44%"&gt;
    &lt;a href="https://www.resilience.org/stories/2020-03-09/the-last-crop-before-the-desert/" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_morocco.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://www.tibettravel.org/tibetan-culture/highland-barley.html" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_seed_tibet.jpg"&gt;&lt;/a&gt;
    &lt;a href="https://www.nationalgeographic.co.uk/travel/2020/05/photo-story-from-barley-fields-to-whisky-barrels-in-rural-scotland" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../figs/barley_seed_scotland_cropped.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:8%; font-size: 15px;"&gt;
    &lt;p style="text-align: center; font-size: 30px; line-height: 1em;"&gt; &lt;strong&gt; Barley across the world &lt;/strong&gt;&lt;/p&gt;
    &lt;p&gt;Kiliskoye (Chelyabinsk, Russia)&lt;/p&gt;
    &lt;p&gt;Marchouch (Rabat, Morocco)&lt;/p&gt;
    &lt;p&gt;Aksum (Tigray, Ethiopia)&lt;/p&gt;
    &lt;p&gt;Salar (Tsetang, Tibet)&lt;/p&gt;
    &lt;p&gt;Expansion of the barley. &lt;/p&gt;
    &lt;p&gt;Turriff (Aberdeenshire, Scotland)&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

## Evolution in real time &amp;rarr; X-rays &amp;rarr; Image Processing

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:41%; color: Navy; font-size: 15px;"&gt;
    &lt;img style="padding: 2px 0 2px 0;" src="../figs/barley_world.jpg"&gt;
    &lt;p style="text-align: center;"&gt;28 accessions around the world&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:50%; color: Navy; font-size: 15px;"&gt;
    &lt;img style="padding: 2px 0 2px 0;" src="../figs/composite_cross_collection_06.svg"&gt;
    &lt;p style="text-align: center;"&gt;58 generations in California&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:51%; color: Navy; font-size: 15px;"&gt;
    &lt;img style="padding: 2px 0 2px 0;" src="../figs/x3000_setup.jpg"&gt;
    &lt;p style="text-align: center;"&gt; Proprietary X-Ray CT scan reconstruction &lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:17.5%; color: Navy; font-size: 15px;"&gt;
    &lt;img src="../figs/S019_L0_1.gif"&gt;
    &lt;p style="text-align: center;"&gt; 975 spikes &lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20.5%; color: Navy; font-size: 15px;"&gt;
    &lt;img src="../figs/S017_L0_seed_10_0.gif"&gt;
    &lt;p style="text-align: center;"&gt; 38,000 seeds &lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

background-image: url("../figs/S017_L0_seed_10_0.gif")
background-size: 85px
background-position: 90% 2%

# Traditional shape descriptors

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:55%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../figs/boxplot_founders_Length.png"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../figs/boxplot_founders_Area.png"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:45%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../figs/boxplot_all_vol_h.png"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../figs/boxplot_all_length_h.png"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../figs/boxplot_all_area_h.png"&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

# Topology: The Euler characteristic

`$$\begin{align} \chi &amp;= \#(\text{Vertices}) - \#(\text{Edges}) + \#(\text{Faces}) \\ &amp;= \#(\text{Connected Components}) - \#(\text{Loops}) + \#(\text{Voids}). \end{align}$$`

&lt;div class="row"&gt;
  &lt;div class="column" style="width:5%"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tda/figs/tetrahedron.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tda/figs/cube.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tda/figs/octahedron.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="width:5%"&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row"&gt;
  &lt;div class="column" style="width:5%"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tda/figs/dodecahedron.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tda/figs/icosahedron.gif"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;p style="padding: 50px 0 0 10px; font-size: 48px; text-align: center; color: #18453B; line-height: 1.25; font-family: 'Yanone Kaffeesatz';"&gt; V - E + F = 2&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="width:5%"&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

background-image: url("../figs/ecc_ver2.gif")
background-size: 750px
background-position: 50% 90%

## Euler Characteristic Curve (ECC)

- Consider a cubical complex `\(X\subset\mathbb{R}^d\)`
- And a unit-length direction `\(\nu\in S^{d-1}\)`

- And the subcomplex containing all cubical cells below height `\(h\)` in the direction `\(\nu\)`
`$$X(\nu)_h =\{\Delta \in X\::\:\langle x,\nu\rangle\leq h\text{ for all }x\in\Delta\}$$`

- The Euler Characteristic Curve (ECC) of direction `\(\nu\)` is defined as the sequence `$$\{\chi(X(\nu)_h)\}_{h\in\mathbb{R}}$$`

---

background-image: url("../figs/ect_ver2.gif")
background-size: 800px
background-position: 50% 88%

## Euler Characteristic Transform (ECT)

- Repeat and concatenate for all possible directions.

$$
`\begin{split}
ECT(X):\; &amp; S^{d-1} \to \mathbb{Z}^{\mathbb{R}}\\
&amp;\nu\mapsto\{\chi(X(\nu)_h)\}_{h\in\mathbb{R}}.
\end{split}`
$$

- [**Theorem** (Turner, Mukherjee, Boyer 2014) &amp; (Curry, Mukherjee, Turner, 2018)](http://arxiv.org/abs/1805.09782):  The ECT is injective with a finite bound of necessary directions.

---

background-image: url("../figs/turner_etal_2014.png")
background-size: 280px
background-position: 99% 1%


# More on the ECT

- Easy to compute: a quick alternating sum.

- *Different* simplicial complexes correspond to *different* ECTs.

- [**Theorem _(ibid)_**](https://arxiv.org/abs/1310.1030): The ECT effectively summarizes all possible information related to shape.

There is elusive math research on computationally efficient reconstruction algorithms:
- [Turner, Mukherjee, Curry (2021)](https://arxiv.org/abs/1805.09782): Finite number of directions
- [Betthauser (2018)](https://people.clas.ufl.edu/peterbubenik/files/Betthauser_Thesis.pdf): 2D reconstruction
- [Fasy, Micka, Millman, Schenfisch, Williams (2022)](https://arxiv.org/abs/1912.12759): 3D reconstruction
- [Crawford, Monod, Chen, Mukherjee, Rabadan (2020)](https://doi.org/10.1080/01621459.2019.1671198): Smooth ECT
- [Jiang, Kurtek, Needham (2020)](https://openaccess.thecvf.com/content_CVPRW_2020/papers/w50/Jiang_The_Weighted_Euler_Curve_Transform_for_Shape_and_Image_Analysis_CVPRW_2020_paper.pdf): Weighted ECT

&lt;img src="../../arabidopsis/figs/ect_col-0_pot0_leaf0.gif" width="500px" style="display: block; margin: auto;" /&gt;

---

background-image: url("../figs/ect_ver2.gif")
background-size: 600px
background-position: 50% 98%

## Topological shape descriptors for barley seeds

.pull-left[
&lt;img src="../figs/pole_directions_p7_m12_crop.jpg" width="250" style="display: block; margin: auto;" /&gt;
]

.pull-right[
- 158 directions

- 16 slices per direction

- Every seed is associated a `\(158\times16=2528\)`-dim vector

- Reduced to 12 dimensions with UMAP
]


---

# Supervised Learning

![](../../demat/figs/supervised_learning.png)

---

background-image: url("../figs/barley_world.jpg")
background-size: 225px
background-position: 98% 2%

# Quantify the shape of barley

**Goal:** Classify the **28** barley accessions solely by grain morphology.

&lt;style type="text/css"&gt;
.tg  {border-collapse:collapse;border-color:#93a1a1;border-spacing:0;margin:0px auto;}
.tg td{background-color:#fdf6e3;border-bottom-width:1px;border-color:#93a1a1;border-style:solid;border-top-width:1px;
  border-width:0px;color:#002b36;font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 5px;
  word-break:normal;}
.tg th{background-color:#657b83;border-bottom-width:1px;border-color:#93a1a1;border-style:solid;border-top-width:1px;
  border-width:0px;color:#fdf6e3;font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;
  padding:10px 5px;word-break:normal;}
.tg .tg-2bhk{background-color:#eee8d5;border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-0pky{border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-gyvr{background-color:#eee8d5;border-color:inherit;font-size:100%;text-align:left;vertical-align:top}
&lt;/style&gt;
&lt;table class="tg"&gt;
&lt;thead&gt;
  &lt;tr&gt;
    &lt;th class="tg-0pky"&gt;Shape descriptors&lt;/th&gt;
    &lt;th class="tg-0pky"&gt;No. of descriptors&lt;/th&gt;
    &lt;th class="tg-0pky"&gt;F1&lt;/th&gt;
  &lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
    &lt;td class="tg-2bhk"&gt;Traditional&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;11&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;0.55 &amp;plusmn; 0.019&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td class="tg-0pky"&gt;Topological&lt;/td&gt;
    &lt;td class="tg-0pky"&gt;12&lt;/td&gt;
    &lt;td class="tg-0pky"&gt;0.74 &amp;plusmn; 0.016&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td class="tg-2bhk"&gt;Combined&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;23&lt;/td&gt;
    &lt;td class="tg-2bhk"&gt;&lt;strong&gt;0.86 &amp;plusmn; 0.010&lt;/strong&gt;&lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

### What does topology actually measure?

.pull-left[
&lt;img src="../figs/discerning_directions.png" width="225" style="display: block; margin: auto;" /&gt;
]

.pull-right[
![](../figs/arrow_seed_09_0.gif)
]

---

# Predicting with Semi-Supervised Learning

![](../../demat/figs/semisupervised_learning.png)

---

# Prediction into semi-supervised territory

- Train an SVM with 100% of the founders `\((F_0)\)`

- Classify the progeny `\((F_{18}\text{ and }F_{58})\)` to detect genotype enrichment

.pull-left[
![](../figs/histogram_combined_gen0_d158_T16_topounscaled_vert.svg)
]

.pull-right[
![](../figs/histogram_combined_gen7_d158_T16_topounscaled_vert.svg)
]

---

background-image: url("../figs/comparison_lines_combined_topounscaled_d158_T16_horz.png")
background-size: 900px
background-position: 50% 70%

## Observe shape distribution across lines

---

background-image: url("../figs/amezquita_etal_2021.png")
background-size: 300px
background-position: 98% 2%


# Summary

.left-column[
&lt;img src="../figs/S017_L3_1.gif" width="150" style="display: block; margin: auto;" /&gt;&lt;img src="../figs/ecc_X.gif" width="150" style="display: block; margin: auto;" /&gt;
]

.right-column[
- **TDA** captures **nuanced** morphological information that is not obvious to the naked eye.

- We get the best results by **combining** both traditional and topological morphological descriptors.

- We can tell **similar** population genetic dynamics by using combined **morphological** descriptors.

- Seed phenotyping can be **cheaper** than genotyping.

- This is one of the **first** applications of the ECT to real-world data.

- First application of the ECT to **plant biology** specifically.
]

---

class: center, inverse, middle

# 2. If life gives you lemons...

## Modeling citrus oil gland distribution

### Comparing limes to oranges with directional statistics

&lt;img src="../../citrus/figs/ppp3.png" width="400" style="display: block; margin: auto;" /&gt;

---

## Think of citrus as lego blocks

![](../../citrus/figs/citrus_family_tree_1.jpg)
&lt;p style="font-size: 10px; text-align: right; color: Grey;"&gt; Credits: &lt;a href="https://doi.org/10.1038/nature25447"&gt;Wu &lt;em&gt;et al.&lt;/em&gt; (2018)&lt;/a&gt;&lt;/p&gt;

---

## X-rays &amp;rarr; Image Processing &amp;rarr; Oil glands

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:37%; color: Navy; font-size: 15px;"&gt;
    &lt;img style="padding: 2px 0 2px 0;" src="../../citrus/crc_pics/crc_diversity.jpg"&gt;
    &lt;p style="text-align: center;"&gt;UCR Collaboration&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:37%; color: Navy; font-size: 15px;"&gt;
    &lt;img style="padding: 2px 0 2px 0;" src="../../citrus/crc_pics/citrus_xrayct_scanning.jpeg"&gt;
    &lt;p style="text-align: center;"&gt;3D X-Ray CT scan&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:25%; color: Navy; font-size: 15px;"&gt;
    &lt;img src="../../citrus/figs/lemon_development.jpg"&gt;
    &lt;p style="text-align: center;"&gt;Fruit development&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row" style="color: black; font-size: 20px; font-family: 'Yanone Kaffeesatz'; margin: 0 auto;"&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../citrus/figs/SR01_L01_black_endocarp.gif"&gt;
    &lt;p style="text-align: center;"&gt; Endocarp &lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../citrus/figs/SR01_L01_black_rind.gif"&gt;
    &lt;p style="text-align: center; padding:5px 0;"&gt; Rind &lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../citrus/figs/SR01_L01_black_exocarp.gif"&gt;
    &lt;p style="text-align: center;"&gt; Exocarp &lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../citrus/figs/SR01_L01_black_oil_glands.gif"&gt;
    &lt;p style="text-align: center;"&gt; Oil glands&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;padding: 10px 0 0 20px;"&gt;
    &lt;p&gt;
      &lt;li&gt;53 different citrus species&lt;/li&gt;&lt;br&gt;
      &lt;li&gt;All fundamental lego blocks and several relatives and hybrids of interest&lt;/li&gt;&lt;br&gt;
      &lt;li&gt;166 individual fruits&lt;/li&gt;
    &lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

background-image: url("../../citrus/figs/lambert_equal_area_N.gif")
background-size: 150px
background-position: 98% 1%

class: middle

## Citrus modeling: Sour and sweet orange

&lt;img src="../../citrus/figs/SR01_CRC3289_12B-19-9_L00_lambproj.jpg" width="800" style="display: block; margin: auto;" /&gt;&lt;img src="../../citrus/figs/SW03_CRC1241-B_12B-4-3_L00_lambproj.jpg" width="800" style="display: block; margin: auto;" /&gt;

---

# Citrus projections

![](../../citrus/figs/citrus_comparison_panel_alt.jpg)

---

background-image: url("../../citrus/ellipsoids/GarciaPortugues_etal_2020a.png")
background-size: 200px
background-position: 99% 1%

# Uniform and rotationally symmetric spherical distributions

![](../../citrus/figs/rotasym_comparison_panel_alt.jpg)

- A battery of tests reveal that citrus oil glands are neither uniformly nor rotationally symmetrically distributed.

---

background-image: url("../../citrus/ellipsoids/GarciaPortugues_2013.png")
background-size: 150px
background-position: 95% 3%

# Spherical Kernel Density Estimators

![](../../citrus/figs/WR19_CRC3842_18B-18-5_L00_emi_kde_N.jpg)

- Oil glands of a papeda
- Red arrows indicate the most significant gradient values for the density function
- Concentration parameter `\(k\)` is chosen to be optimal

---

background-image: url("../../citrus/figs/ppp3.png")
background-size: 225px
background-position: 95% 3%

# Summary

.left-column[

&lt;img src="../../citrus/crc_pics/citrus_amblycarpa.jpg" width="150" style="display: block; margin: auto;" /&gt;&lt;img src="../../citrus/figs/SR01_L01_black_exocarp.gif" width="150" style="display: block; margin: auto;" /&gt;&lt;img src="../../citrus/figs/SW03_CRC1241-B_12B-4-3_L00_lambproj_v.jpg" width="150" style="display: block; margin: auto;" /&gt;

]

.right-column[
- It is possible to **compare** ~~apples~~ limes to oranges, despite diversity in shapes and sizes.

- We can define the non-parametrical **distribution** of oil glands for most of the citrus.

- **First** time citrus shape has been examined using X-ray CT scans.

- AFAIK, this is also the **first** time directional statistics have been **applied** to plant biology.
]

---

class: inverse, center, middle

# 3. All about cracks, kernels, and domestication

## In a nutshell

&lt;img src="https://upload.wikimedia.org/wikipedia/commons/3/39/Benutzung_eines_Nussknackers.gif" width="200" style="display: block; margin: auto;" /&gt;

---

# The wal(nu)tzing nutcracker!

&lt;div class="row" style="font-family: 'Yanone Kaffeesatz'; margin: 0 auto; font-size:22px"&gt;
  &lt;div class="column" style="max-width:25%;"&gt;
    &lt;img src="../../walnuts/figs/IMGA04011.jpg"&gt;
    &lt;p style="text-align: center;"&gt;148 accessions&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:42%;"&gt;
    &lt;img src="../../barley/figs/x3000_setup.jpg"&gt;
    &lt;p style="text-align: center;"&gt;1301 walnuts scanned&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%;"&gt;
    &lt;img src="../../walnuts/figs/pat_walnut_crack_detail4.jpg"&gt;
    &lt;p style="text-align: center;"&gt;Qualitative data for crackability&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row" style="font-size: 22px; font-family: 'Yanone Kaffeesatz'; margin: 0 auto;"&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../walnuts/figs/2014SBa_R5_T81_shell_1.gif"&gt;
    &lt;p style="text-align: center; margin: 0.5ex;"&gt;Shell&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../walnuts/figs/2014SBa_R1_T3_001_air.gif"&gt;
    &lt;p style="text-align: center;margin: 0.5ex; "&gt;Air&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../walnuts/figs/2014SBa_R1_T3_001_vein.gif"&gt;
    &lt;p style="text-align: center;margin: 0.5ex; "&gt;Packing tissue&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%;"&gt;
    &lt;img src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif"&gt;
    &lt;p style="text-align: center;margin: 0.5ex;"&gt;Kernel&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%; padding: 0 0 0 5px"&gt;
    &lt;p style="margin:5px 0 0 0; text-align:left"&gt;
      &amp;#9656;&amp;ensp;Individuals separated manually&lt;br&gt;
      &amp;#9656;&amp;ensp;Tissues separated with watershed segmentation&lt;br&gt;
      &amp;#9656;&amp;ensp;38 traditional shape and size descriptors
    &lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

# The power of allometry: `\(\;y = e^bV_N^m\)`

For most living organisms, different tissues grow at different rates with respect to each other.

Growth does **not** scale linearly but following a power law.

&lt;div class="row" style="color: black; font-size: 20px; font-family: 'Yanone Kaffeesatz'; margin: 0 auto;"&gt;
  &lt;div class="column" style="max-width:33%;"&gt;
    &lt;img src="../../walnuts/figs/nut_convex_volume_vs_nut_volume.png"&gt;
    &lt;p style="padding: 0 2ex 0 0;"&gt;Walnut changes dramatically after its diameter is larger than 1.6cm&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%;"&gt;
    &lt;img src="../../walnuts/figs/air_volume_vs_nut_volume.png"&gt;
    &lt;p style="margin: 3px 0 0 0; padding: 0 2ex 0 0;"&gt;If nut volume increases by 2x, then air volume increases by 2.3x&lt;/p&gt;
    &lt;p style="padding: 0 2ex 0 0;"&gt;Walnut diameter is below 16cm&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:33%;"&gt;
    &lt;img src="../../walnuts/figs/kernel_volume_vs_nut_volume.png"&gt;
    &lt;p style="margin: 1.1ex 0 0 0; padding: 0 2ex 0 0;"&gt;If nut volume increases by 2x, then kernel volume only increases by 1.8x&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

There are biophysical constraints at play in walnut development

---

## Correlation with traits of interest

![](../../walnuts/figs/combination_hbar_correlation.svg)

.pull-left[
**Walnuts with more kernel filling ratio have:**

- Less air filling ratio
- Smaller volume
- Smoother shells
- Kernel overall shape is more convex 
- But it also has numerous deep but thin grooves.
] 

.pull-right[
**Walnuts with thinner shells have:**

- Less total shell volume and shell ratio
- More air filling ratio
- Smoother shells
]

---

&lt;h3 style="line-height:0"&gt;Descriptors that explain the most crackability variance&lt;/h3&gt;

&lt;img src="../../walnuts/figs/qual_EaseOfRemoval_scatter_2x4.jpg" width="725" style="display: block; margin: auto;" /&gt;&lt;img src="../../walnuts/figs/qual_ShellStrength_scatter_2x4.jpg" width="725" style="display: block; margin: auto;" /&gt;

---

## Can you spot the Earliest Himalayan accession?

&lt;img src="../../walnuts/figs/pat_walnut_crack_crop.jpg" width="500" style="display: block; margin: auto;" /&gt;

This one accession is the most heterozygous wild type of the bunch

---

## The Earliest Himalayan accession breaks the pattern

&lt;img src="../../walnuts/figs/NewStuke_R1_T18_vs_2014SBa_R7_T26.png" width="725" style="display: block; margin: auto;" /&gt;

--

&lt;img src="../../walnuts/figs/qual_85-023-2_scatter.jpg" width="725" style="display: block; margin: auto;" /&gt;

- The kernel of Earliest has deeper, wider cavities
- Does it refer to the cavity located at the proximal end?

---

background-image: url("https://toasterovenlove.com/wp-content/uploads/Stirred-nuts-on-pan.jpg")
background-size: 180px
background-position: 98% 2%

# Summary

.left-column[

&lt;img src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif" width="725" style="display: block; margin: auto;" /&gt;&lt;img src="../../walnuts/figs/kernel_volume_vs_nut_volume.png" width="725" style="display: block; margin: auto;" /&gt;

]

.right-column[

- There are serious **biophysical** constraints that govern the kernel filling ratio.

- Larger walnuts tend to have **less** kernel and **more** air relative to their size.

- Walnuts with thinner shells also have **more** air.

- **Less** packing tissue in general corresponds to better **crackability**.

- The shape of **domestication** is quite **nuanced**: it is a hard nut to crack.

- AFAIK, this is the first time that **allometry** has been studied for walnut fruit.

- AFAIK, first time packing tissue volume and **convexity** indices have been measured.
]

---

class: center, inverse, middle

# 4. Future directions

## Barley scratching the surface

---

# Grains, oranges, and nuts oh my!

.left-column[

&lt;img src="../figs/S017_L3_1.gif" width="100" style="display: block; margin: auto;" /&gt;&lt;img src="../../citrus/figs/SR01_L01_black_exocarp.gif" width="100" style="display: block; margin: auto;" /&gt;&lt;img src="../../walnuts/figs/2014SBa_R5_T81_meat_2.gif" width="100" style="display: block; margin: auto;" /&gt;

]

.right-column[
- Identify specific **molecular markers** corresponding to topological traits of **barley seeds**.
- Develop a high-throughput **seed phenotyping** pipeline for seed morphology.

- Compute a pairwise distance matrix for citrus oil gland **distributions**.
- Use **persistent homology** as an alternative to quantify the distribution of such oil glands.

- Measure the size and volume of the central **cavity** of the walnut kernel.
- Use the ECT or a similar approach to quantify **nuances** present in the kernel.
]


---

background-image: url("https://gradschool.missouri.edu/wp-content/uploads/2019/02/MU_UnitSig_GraduateSchool_rgb_std_horiz.png")
background-size: 330px
background-position: 98% 2%

## Future research at Mizzou

.pull-left[
The shape of maize roots

&lt;img src="../../psd/figs/Maiz_root_Dario.jpeg" width="200" style="display: block; margin: auto;" /&gt;

The shape of dodder movement

&lt;img src="https://i.makeagif.com/media/11-28-2013/qmlphv.gif" width="300" style="display: block; margin: auto;" /&gt;

]

.pull-right[
Dabbing into UAV imagery for sorghum high-throughput phenotyping

![](https://pub.mdpi-res.com/remotesensing/remotesensing-11-02075/article_deploy/html/images/remotesensing-11-02075-ag.png?1569418523)

- David Mendoza Cozatl
- Robert Sharpe
- So-Yon Park
- Andrew Scaboo

]

---

background-image: url("../../tutorials/figs/plantspython22_01.png")
background-size: 225px
background-position: 99% 1%

# Plants &amp; Python _a la_ Mizzou

.pull-left[
![](https://plantsandpython.github.io/PlantsAndPython/_static/plants_python_logo.jpg)
![](../../tutorials/figs/class_foto1.png)
]

.pull-right[
- **Inspired by the current Plants &amp; Python course @ MSU**

- Offered in hybrid format between MSU and UNAM

- **HRT 841**: Foundation in Computational and Plant Science

- **UNAM Temas Selectos**: Bioinformática y Minería de Datos con Python

- Freely available material in jupyter notebook format.

- Material in English and Spanish.
]


---

background-image: url("../../cmse/figs/institutional_logos.jpg")
background-size: 500px
background-position: 95% 1%

class: inverse

## Thank you!

&lt;div class="row" style="margin-top: -25px;"&gt;
  &lt;div class="column" style="max-width:17.5%; font-size: 13px;"&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://i1.rgstatic.net/ii/profile.image/607374528233472-1521820780707_Q128/Elizabeth-Munch.jpg"&gt;
    &lt;p style="text-align: center; color: White"&gt;Liz Munch&lt;br&gt;(MSU)&lt;/p&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://alga.win.tue.nl/images/staff/tim.jpg"&gt;
    &lt;p style="text-align: center; color: White"&gt;Tim Ophelders&lt;br&gt;(Utrecht)&lt;/p&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://publons.com/media/thumbs/academic/photos/252a7116-ee14-4a5e-b44c-7c62ac11e217.png.200x200_q95_crop_detail_upscale.png"&gt;
    &lt;p style="text-align: center; color: White"&gt;Jacob Landis&lt;br&gt;(Cornell)&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:17.5%; font-size: 13px;"&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://www.canr.msu.edu/contentAsset/image/9ae9777d-157c-46e6-9f12-d062ad35671e/fileAsset/filter/Jpeg,Resize,Crop/jpeg_q/80/resize_w/400/crop_x/0/crop_y/45/crop_w/400/crop_h/400"&gt;
    &lt;p style="text-align: center; color: White"&gt;Dan Chitwood&lt;br&gt;(MSU)&lt;/p&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://i1.rgstatic.net/ii/profile.image/926632407748609-1597937790454_Q128/Michelle-Quigley.jpg"&gt;
    &lt;p style="text-align: center; color: White"&gt;Michelle Quigley&lt;br&gt;(MSU)&lt;/p&gt;
    &lt;img style="padding: 0 0 0 0;" src="https://www.plantsciences.ucdavis.edu/sites/g/files/dgvnsk1736/files/styles/sf_profile/public/images/person/pjbrown_1.jpg?h=9707347b&amp;itok=gDYXMHJj"&gt;
    &lt;p style="text-align: center; color: White"&gt;Pat Brown&lt;br&gt;(UC Davis)&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:13%; font-size: 13px;"&gt;
  &lt;img style="padding: 0 0 0 0;" src="https://plantbiology.ucr.edu/sites/g/files/rcwecm1001/files/styles/scale_225/public/Koenig.jpg?itok=P92Ow61p"&gt;
  &lt;p style="text-align: center; color: White"&gt;Dan Koenig&lt;br&gt;(UC Riverside)&lt;/p&gt;
  &lt;img style="padding: 0 0 0 0;" src="https://iigb.ucr.edu/sites/g/files/rcwecm5716/files/styles/scale_225/public/Seymour.jpg?itok=pnUK_orh"&gt;
  &lt;p style="text-align: center; color: White"&gt;Danelle Seymour&lt;br&gt;(UC Riverside)&lt;/p&gt;
  &lt;/div&gt;
  &lt;div class="column" style="width:10%; font-size: 24px;"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:40%; font-size: 24px; line-height:1.25"&gt;
  &lt;p style="text-align: center; color: White"&gt;&lt;strong&gt;Email&lt;/strong&gt;&lt;/p&gt;
  &lt;p style="text-align: center; color: Yellow"&gt;amezqui3@msu.edu&lt;/p&gt;
  &lt;p style="text-align: center; color: White"&gt;&lt;strong&gt;Slides&lt;/strong&gt;&lt;/p&gt;
  &lt;p style="text-align: center; color: Yellow"&gt;bit.ly/mu_22&lt;/p&gt;
  &lt;p style="text-align: center; color: White"&gt;&lt;strong&gt;References&lt;/strong&gt;&lt;/p&gt;
  &lt;p style="text-align: center; color: White; font-size: 12px; text-align:left"&gt;
    &lt;strong&gt;E.A.&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2022) The shape of aroma: measuring and modeling citrus oil gland distribution. &lt;a href="https://doi.org/10.1002/ppp3.10333" target="_blank"&gt;DOI: 10.1002/ppp3.10333&lt;/a&gt;
  &lt;/p&gt;
  &lt;p style="text-align: center; color: White; font-size: 12px; text-align:left"&gt;
    &lt;strong&gt;E.A.&lt;/strong&gt; &lt;em&gt;et al.&lt;/em&gt; (2021) Measuring hidden phenotype: Quantifying the shape of barley seeds using the Euler Characteristic Transform. &lt;a href="https://doi.org/10.1093/insilicoplants/diab033" target="_blank"&gt;DOI: 10.1093/insilicoplants/diab033&lt;/a&gt;
  &lt;/p&gt;
  &lt;p style="text-align: center; color: White; font-size: 12px; text-align:left"&gt;
    VanBuren, R. &lt;em&gt;et al.&lt;/em&gt; (2022) Plants &amp; Python: A series of lessons in coding, plant biology, computation, and bioinformatics. &lt;a href="https://doi.org/10.1093/plcell/koac187" target="_blank"&gt;DOI: 10.1093/plcell/koac187&lt;/a&gt;
  &lt;/p&gt;
    &lt;p style="text-align: center; color: White; font-size:15px; line-height:1.1"&gt;Slides made in &lt;a href="https://bookdown.org/yihui/rmarkdown/xaringan.html" target="_blank"&gt;xaringan&lt;/a&gt;.&lt;/p&gt;
  &lt;/div&gt;
&lt;/div&gt;

---

class: inverse, center, middle

# Questions?

![](https://64.media.tumblr.com/895f079d8cc1128b5ce3ca7aacf57389/9ff35c052f5d8802-16/s500x750/55486cc6b04f5f7192d0002707ca8b7e550e4db8.gifv)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="../../js/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
