<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Going between math, data science, plant biology, and education</title>
    <meta charset="utf-8" />
    <meta name="author" content="Erik Amézquita — Division of Plant Science &amp; Technology Department of Mathematics University of Missouri —" />
    <meta name="date" content="2025-08-18" />
    <script src="libs/header-attrs-2.27/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../css/mu.css" type="text/css" />
    <link rel="stylesheet" href="../../css/gallery.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Going between math, data science, plant biology, and education
]
.subtitle[
## Journeys into interdisciplinarity
]
.author[
### Erik Amézquita<br>—<br>Division of Plant Science &amp; Technology<br>Department of Mathematics<br>University of Missouri<br>—
]
.date[
### 2025-08-18
]

---


background-image: url("../../demat/figs/fam9_3.png")
background-size: 100px
background-position: 98% 2%

# About me: From MX to MI to MO at MU

## I work across multiple disciplines and countries

.left-column[
![](../../img/dvdy216-toc-0001-m.jpg)
]

.right-column[
- 2013 - 2018 : Licenciatura (Bachelor): Mathematics @ Universidad de Guanajuato and CIMAT. Thesis focused on Topological Data Analysis applied to archaeology.

- 2018 - 2023 : PhD: Computational Mathematics, Science, and Engineering @ Michigan State University. Dissertation: Exploring the mathematical shape of plants. **Came for the math. Stayed for the plants.**

- 2023 - Present : PFFIE Postdoctoral Fellow @ Division of Plant Science &amp; Technology (80%) / Department of Mathematics (20%) at MU.
]

&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;&lt;br&gt;

---

# My work: Crossing and merging bridges

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:21%"&gt;
    &lt;a href="https://doi.org/10.1002/ppj2.20095" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../walnuts/figs/meat_2_05.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:20%"&gt;
    &lt;a href="https://doi.org/10.1007/s00299-024-03337-1" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../cuscuta/figs/4pm_rep7_plant2_v09_0401_h.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:21%"&gt;
    &lt;a href="https://doi.org/10.1002/ppj2.20095" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../mcarto/figs/Infected_Cells_05G216000_red_05G092200_green.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:27%"&gt;
    &lt;a href="https://doi.org/10.1371/journal.pone.0284820" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../nasrin/figs/lung_fpkm_meancorr_eps1.0e+06_r40_g40.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="row"&gt;
  &lt;div class="column" style="max-width:43%"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/myzou_plnt_sci_2500.png"&gt;
    &lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/mizzou_math_drp.png"&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:30%"&gt;
    &lt;a href="https://doi.org/10.1101/2022.10.15.512190" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../tutorials/figs/ipg_network_r.jpg"&gt;&lt;/a&gt;
  &lt;/div&gt;
  &lt;div class="column" style="max-width:25%"&gt;
    &lt;a href="https://doi.org/10.1101/2022.10.15.512190" target="_blank"&gt;&lt;img style="padding: 0 0 0 0;" src="../../tda/figs/107-tulare.png"&gt;&lt;/a&gt;
  &lt;/div&gt;
&lt;/div&gt;

&lt;div class="list" style="font-size: 14px; text-align: left;"&gt;
    &lt;ul&gt;
      &lt;li&gt;M Bentelspacher, &lt;strong&gt;EJA&lt;/strong&gt;, S Adhikari, J Barros, SY Park (2024) "The early dodder gets the host: Decoding the coiling patterns of Cuscuta campestris with automated image processing". &lt;em&gt;Plant Cell Reports&lt;/em&gt;.&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;EJA&lt;/strong&gt;, MY Quigley, PJ Brown, E Munch, DH Chitwood (2024) "Allometry and volumes in a nutshell: Analyzing walnut morphology using three-dimensional X-ray computed tomography". &lt;em&gt;The Plant Phenome J&lt;/em&gt;.&lt;/li&gt;
      &lt;li&gt;&lt;strong&gt;EJA&lt;/strong&gt;, F Nasrin, KM Storey, M Yoshizawa (2023) "Genomics data analysis via spectral shape and topology." &lt;em&gt;PLOS One&lt;/em&gt;&lt;/li&gt;
      &lt;li&gt;SA Cervantes-Pérez &lt;em&gt;et al.&lt;/em&gt; (2024) "Tabula Glycine: The whole-soybean single-cell resolution transcriptome atlas." Submitted.&lt;/li&gt;
      &lt;li&gt;Z Ji, &lt;strong&gt;EJA&lt;/strong&gt;, L Newton, DH Chitwood, AM Thompson (2024) "From hand measurements to high throughput phenotyping: understanding maize canopy structure and predicting yield." Submitted.&lt;/li&gt;
    &lt;/ul&gt;
&lt;/div&gt;

---


## PLNT_SCI 2500: Python taught for life sciences

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-10_In-Class_NumPyDataAnalysis2-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---

## PLNT_SCI 2500: Active learning, example driven

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-13_In-Class_Regression-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---

## PLNT_SCI 2500: With data from DPST faculty!

&lt;p align="center"&gt;
&lt;iframe width="800" height="550" src="../../tutorials/plnt_2500/Day-20_In-Class_AdvancedPlotting-INSTRUCTOR.html" title="Day10"&gt;
&lt;/iframe&gt;
&lt;/p&gt;

---

background-image: url("https://www.biorxiv.org/content/biorxiv/early/2022/09/09/2022.09.07.506951/F1.large.jpg?width=800&amp;height=600&amp;carousel=1")
background-size: 325px
background-position: 1% 60%

# Discussion and future goals

**PLNT_SCI 2500: Data Science for Life Sciences I will be the first course for the emerging Data Science for Life Sciences Certificate**

- Incorporate examples from outside Plant Science to appeal to more students

- Data Science for Life Sciences II will build on top more Data Science topics:
    - Supervised and unsupervised classification
    - Clustering algorithms
    - Non-linear regressions
    - Network analyses
    - Statistical paradoxes to be wary of
    
**Professional Goal: Make the teaching of data science in life sciences a scientific endeavor in itself**

- Collaborate with colleagues with experience in Education research.

- Motivate students to tackle *and publish* a single research problem as a unit.

## Get in touch: ``eah4d@missouri.edu``
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="../../js/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
